---
title: "Factorial experiment with dox treatment"
output: html_document
---

[Link to GitHub issue](https://github.com/datacamp/courses-differential-expression-with-limma/issues/3#issuecomment-334592606)

Zhang et al., 2012:

* title: Identification of the molecular basis of doxorubicin-induced cardiotoxicity
* question: What is the role of the topoisomerase Top2b in the molecular mechanism of doxorubicin-induced cardiotoxicity?
* species: Mus musculus
* tissue: heart
* groups: 2 x 2 x 2 (genotype x treatment x timepoint)
* n per group: 3

[Link to PubMed](https://www.ncbi.nlm.nih.gov/pubmed/23104132)

[Link to GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE40289)

```{r packages, message=FALSE}
library("Biobase")
library("dplyr")
library("GEOquery")
options('download.file.method.GEOquery' = 'wget')
library("ggplot2")
library("limma")
library("stringr")
library("UpSetR")
```

```{r download}
eset_file <- "factorial-dox.rds"
if (file.exists(eset_file)) {
  eset <- readRDS(eset_file)
} else {
  eset <- getGEO("GSE40289", GSEMatrix = TRUE, getGPL = TRUE)[[1]]
  saveRDS(eset, file = "factorial-dox.rds")
}
```

Explore the metadata columns.

```{r}
# genotype
table(pData(eset)$characteristics_ch1.3)
# treatment
table(pData(eset)$characteristics_ch1.4)
# timepoint
levels(pData(eset)$description)
```

Create nice phenotype data columns.

```{r}
pdata_orig <- pData(eset)
pdata_new <- data.frame(title = pdata_orig$title, stringsAsFactors = FALSE)
pdata_new <- mutate(pdata_new,
                   genotype = ifelse(grepl("WT", title), "WT", "Tob2b"),
                   treatment = ifelse(grepl("PBS", title), "PBS", "Dox"),
                   timepoint = ifelse(grepl("16hr", title), "16hr", "72hr"))
pData(eset) <- pdata_new
pdata_new
```

Create nice feature data columns.

```{r}
fdata_orig <- fData(eset)
fdata_new <- fdata_orig %>%
  mutate(chr = str_split_fixed(CHROMOSOMAL_LOCATION, ":", 2)[, 1],
         pos = str_split_fixed(CHROMOSOMAL_LOCATION, ":", 2)[, 2],
         start = str_split_fixed(pos, "-", 2)[, 1],
         end = str_split_fixed(pos, "-", 2)[, 2]) %>%
  select(probe = NAME,
         chr,
         start,
         end,
         symbol = GENE_SYMBOL,
         ensembl = ENSEMBL_ID,
         refseq = GB_ACC,
         entrez = GENE_ID,
         name = GENE_NAME)
fData(eset) <- fdata_new
```

Confirm Top2b expression levels lower in knockout mice.

```{r}
top2b <- exprs(eset)[fData(eset)$symbol == "Top2b", ]
boxplot(top2b ~ eset$genotype + eset$treatment)
```

Using ggplot2.

```{r}
top2b <- data.frame(pData(eset),
                    top2b = exprs(eset)[fData(eset)$symbol == "Top2b", ])
ggplot(top2b, aes(x = treatment, y = top2b, fill = genotype)) +
  geom_boxplot()
```


View gene expression distributions.

```{r}
boxplot(log(exprs(eset)))
plotDensities(log(exprs(eset)), legend = FALSE)
```

View sample relationships with MDS.

```{r}
plotMDS(eset, labels = pData(eset)$genotype)
plotMDS(eset, labels = pData(eset)$treatment)
plotMDS(eset, labels = pData(eset)$timepoint)
```

Dimension 1 is timepoint. Dimension 2 is genotype.

View sample relationships with PCA.

```{r}
pca <- prcomp(t(exprs(eset)), scale. = TRUE)
plot(pca)
plot(pca$x[, 1], pca$x[, 2], col = "white")
text(pca$x[, 1], pca$x[, 2], labels = pData(eset)$genotype)
plot(pca$x[, 1], pca$x[, 2], col = "white")
text(pca$x[, 1], pca$x[, 2], labels = pData(eset)$treatment)
plot(pca$x[, 1], pca$x[, 2], col = "white")
text(pca$x[, 1], pca$x[, 2], labels = pData(eset)$timepoint)
```

PC1 is timepoint. PC2 is genotype.

View sample relationships with hierachical clustering.

```{r}
cor_mat <- cor(exprs(eset))
rownames(cor_mat) <- paste(pData(eset)$genotype,
                           pData(eset)$treatment,
                           pData(eset)$timepoint)
h <- hclust(dist(cor_mat))
plot(h)
```

Remove 72 hour timepoint.

```{r remove-72hr}
eset <- eset[, eset$timepoint == "16hr"]
dim(eset)
```

View gene expression distributions.

```{r}
boxplot(log(exprs(eset)))
plotDensities(log(exprs(eset)), legend = FALSE)
```

View sample relationships with MDS.

```{r}
plotMDS(eset, labels = paste(pData(eset)$genotype, pData(eset)$treatment, sep = "."))
```

Dimension 1 is genotype. Dimension 2 is less clear. Maybe Dox in WT.

View sample relationships with PCA.

```{r}
pca <- prcomp(t(exprs(eset)), scale. = TRUE)
plot(pca)
plot(pca$x[, 1], pca$x[, 2], col = "white")
text(pca$x[, 1], pca$x[, 2], labels = pData(eset)$genotype)
plot(pca$x[, 1], pca$x[, 2], col = "white")
text(pca$x[, 1], pca$x[, 2], labels = pData(eset)$treatment)
plot(pca$x[, 1], pca$x[, 2], col = "white")
text(pca$x[, 1], pca$x[, 2],  labels = paste(pData(eset)$genotype,
                                             pData(eset)$treatment,
                                             sep = "."))
```

PC1 is genotype. PC2 is effect of dox treatment in WT only.

View sample relationships with hierachical clustering.

```{r}
exp <- exprs(eset)
colnames(exp) <- paste(pData(eset)$genotype,
                       pData(eset)$treatment, sep = ".")
cor_mat <- cor(exp)
h <- hclust(dist(cor_mat))
plot(h)
```

That's strange. Bigger effect of Dox in WT, but do not cluster as nicely.

Design.

```{r}
single_factor <- paste(pData(eset)$genotype,
                       pData(eset)$treatment,
                       sep = ".")
design <- model.matrix(~0 + single_factor)
colnames(design) <- sub("single_factor", "", colnames(design))
colSums(design)
```

Fit model.

```{r}
fit <- lmFit(eset, design)
cont_mat <- makeContrasts(DoxWT = WT.Dox - WT.PBS,
                          DoxTop = Tob2b.Dox - Tob2b.PBS,
                          Interaction = (Tob2b.Dox - Tob2b.PBS) -
                                        (WT.Dox - WT.PBS),
                          levels = design)
fit2 <- contrasts.fit(fit, contrasts = cont_mat)
fit2 <- eBayes(fit2)
results <- decideTests(fit2)
```

View results.

```{r}
vennDiagram(results)
upset(as.data.frame(abs(results)))
```

Similar to the results from the paper, there is less of an effect of dox in the Top2b null mice.

KEGG enrichment

```{r}
enrich_kegg_wt <- kegga(fit2, coef = "DoxWT",
                        geneid = fit2$genes$entrez, species = "Mm")
topKEGG(enrich_kegg_wt)
enrich_kegg_inter <- kegga(fit2, coef = "Interaction",
                        geneid = fit2$genes$entrez, species = "Mm")
topKEGG(enrich_kegg_inter)
```

GO enrichment

```{r}
enrich_go_wt <- goana(fit2, coef = "DoxWT",
                      geneid = fit2$genes$entrez, species = "Mm")
topGO(enrich_go_wt, ontology = "BP")
enrich_go_inter <- goana(fit2, coef = "Interaction",
                         geneid = fit2$genes$entrez, species = "Mm")
topGO(enrich_go_inter, ontology = "BP")
```
