<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="John Blischak" />

<meta name="date" content="2018-02-10" />

<title>Introduction to differential expression analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DE analysis with limma</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/dc-bioc-limma">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Introduction to differential expression analysis</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2018-02-10</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-02-16</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 980afe1</p>
<p>Cover the basics of the biology and the statistics.</p>
<pre class="r"><code>gexp &lt;- readRDS(&quot;../data/ch01.rds&quot;)</code></pre>
<div id="what-is-differential-expression-video" class="section level2">
<h2>What is differential expression? (Video)</h2>
<p>Describe a simple model of gene regulation (transcription factors, DNA, RNA, protein) and the functional genomics technologies used to measure the various features. Focus on the type of data they return (continuous vs. discrete).</p>
</div>
<div id="functional-genomics-technologies-multiplechoiceexercise" class="section level2">
<h2>Functional genomics technologies (MultipleChoiceExercise)</h2>
<p>Choose which is true. The options will be various combinations of genomics technologies, what they measure, and the type of data they return. For example, “Microarrays measure RNA levels and return continuous data. ChIP-seq measures transcription factor binding to DNA and returns discrete data.” is true.</p>
</div>
<div id="visualize-one-differentially-expressed-gene" class="section level2">
<h2>Visualize one differentially expressed gene</h2>
<p>Visualize one differentially expressed gene with boxplots (geom_boxplot).</p>
<pre class="r"><code>library(&quot;dplyr&quot;)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(&quot;tidyr&quot;)
library(&quot;stringr&quot;)
single_gene &lt;- gexp %&gt;% as.data.frame %&gt;%
  slice(34) %&gt;%
  gather(key = &quot;status&quot;, value = &quot;gene&quot;) %&gt;%
  mutate(status = str_extract(status, &quot;[a-z]*&quot;))</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)
ggplot(single_gene, aes(x = status, y = gene)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-many-differentially-expressed-genes" class="section level2">
<h2>Visualize many differentially expressed genes</h2>
<p>Visualize many differentially expressed genes with a heatmap (stats::heatmap).</p>
<pre class="r"><code>dim(gexp)</code></pre>
<pre><code>[1] 50  6</code></pre>
<pre class="r"><code>head(gexp)</code></pre>
<pre><code>           con1     con2     con3   treat1   treat2   treat3
gene01 13.74914 16.82282 13.76740 16.97906 16.85671 16.03717
gene02 16.40848 14.02148 15.98390 15.80187 20.92417 17.07906
gene03 11.78808 15.31832 11.56220 13.34076 19.05473 14.08600
gene04 20.28549 12.11155 15.97373 12.71156 14.50692 14.75665
gene05 16.93984 20.25039 13.98756 22.97787 18.19122 36.07593
gene06 13.48818 21.89078 21.25145 12.72401 23.44257 19.38827</code></pre>
<pre class="r"><code>heatmap(gexp)</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="what-is-a-linear-model-video" class="section level2">
<h2>What is a linear model? (Video)</h2>
<p>Review the basics of multiple linear regression (dependent vs independent variables, continuous vs discrete covariates, intercept term, error term). Demonstrate how to specify equations in R (e.g. <code>lm</code> and <code>model.matrix</code>).</p>
<p><strong>Transcript:</strong> As you just visualized, differential expression describes the situation in which a gene has a different mean expression level between conditions. While some gene expression patterns are easily diagnosed as differential expression or not from a quick visualization, you also saw some examples that were more ambiguous. Furthermore, you need a method that is more robust than a quick visual inspection and also scales to thousands of genes. For this you will use the tools of statistical inference to determine if the difference in mean expression level is larger than that expected by chance. Specifically, you will use linear models to perform the hypothesis tests. Linear models are an ideal choice for genomics experiments because their flexibility and robustness to assumptions allow you to conveniently model data from various study designs and data sources.</p>
<p>You should have already been introduced to linear models, for example in a DataCamp course such as Correlation and Regression, or in an introductory statistics course. Here I’ll review the terminology we will use in the remainder of the course, how to specify a linear model in R, and the intuition behind linear models.</p>
<p><span class="math display">\[ Y =  \beta_0 + \beta_1 X_1 + \epsilon \]</span></p>
<p>In this equation of a linear model, Y is the response variable. It must be a continuous variable. In the context of differential expression, it is a relative measure of either RNA or protein expression level for one gene. <span class="math inline">\(X_1\)</span> is an explanatory variable, which can be continuous or discrete, for example, control group versus treatment, or mutant versus wild type. <span class="math inline">\(\beta_1\)</span> quantifies the effect of the explanatory variable on the response variable. Furthermore, we can add additional explanatory variables to the equation for more complicated experimental designs. Lastly, models the random noise in the measurements.</p>
<p>In R, you specify a linear model with the function <code>lm</code>. This uses R’s formula syntax. On the left is the object that contains the response variable, and to the right of the tilde are the objects that contain the explanatory variable.</p>
<pre class="r"><code>lm(y ~ x1)</code></pre>
<p>A second explanatory variable can be added with a plus sign.</p>
<p><span class="math display">\[ Y =  \beta_0 + \beta_1 X_1 + \beta_2 X_2 + \epsilon \]</span></p>
<pre class="r"><code>lm(y ~ x1 + x2)</code></pre>
<p>Building some intuition.</p>
<pre class="r"><code>library(&quot;broom&quot;)
library(&quot;cowplot&quot;)
library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_classic(base_size = 16))

# Simulate a linear regression.
#
# n = sample size
# effect = slope
# error = standard deviation of distribution of residuals
# seed = seed for random number generator
#
# Returns a data.frame with the following columns:
#
# x          Explanatory variable
# y          Response variable
# y_bar      Mean of response variable
# intercept  Intercept of least squares regression line
# slope      Slope of least squares regression line
# y_hat      Fitted values
# fstat      F-statistic
# ss_exp     Explained sum of squares
# ss_res     Residual sum of squares (noise)
sim_lm &lt;- function(n, effect, error, seed = 1) {
  stopifnot(is.numeric(n), n &gt; 0, length(n) == 1)
  stopifnot(is.numeric(effect), length(effect) == 1)
  stopifnot(is.numeric(error), error &gt; 0, length(error) == 1)
  stopifnot(is.numeric(seed), length(seed) == 1)
  
  set.seed(seed)
  x = runif(n, min = -25, max = 25)
  y = x * effect + rnorm(n, sd = error)
  y_bar = mean(y)
  mod &lt;- lm(y ~ x)
  coefs &lt;- coef(mod)
  intercept &lt;- coefs[1]
  slope &lt;- coefs[2]
  y_hat = fitted(mod)
  anova_tidy &lt;- tidy(anova(mod))
  fstat &lt;- anova_tidy$statistic[1]
  ss &lt;- anova_tidy$sumsq
  ss_exp &lt;- ss[1]
  ss_res &lt;- ss[2]
  stopifnot(ss_exp - sum((y_hat - y_bar)^2) &lt; 0.01)
  stopifnot(ss_res - sum(residuals(mod)^2) &lt; 0.01)
  return(data.frame(x, y, y_bar, intercept, slope, y_hat, fstat, ss_exp, ss_res,
                    row.names = 1:n))
}

# Visualize the residual sum of squares
plot_ss_res &lt;- function(d) {
  ggplot(d, aes(x = x, y = y)) +
    geom_point() +
    geom_abline(aes(intercept = intercept, slope = slope)) +
    geom_linerange(aes(ymin = y, ymax = y_hat), color = &quot;red&quot;,
                   linetype = &quot;dashed&quot;) +
    geom_text(aes(x = quantile(x, 0.25), y = quantile(y, 0.75),
                  label = round(ss_res, 2)), color = &quot;red&quot;) +
    labs(title = &quot;Residual sum of squares (noise)&quot;) +
    ylim(-60, 60)
}

# Visualize the explained sum of squares
plot_ss_exp &lt;- function(d) {
  ggplot(d, aes(x = x, y = y)) +
    geom_abline(aes(intercept = intercept, slope = slope)) +
    geom_hline(aes(yintercept = y_bar)) +
    geom_linerange(aes(ymin = y_hat, ymax = y_bar), color = &quot;blue&quot;,
                   linetype = &quot;dashed&quot;) +
    geom_text(aes(x = quantile(x, 0.25), y = quantile(y, 0.75),
                  label = round(ss_exp, 2)), color = &quot;blue&quot;) +
    labs(title = &quot;Explained sum of squares&quot;) +
    ylim(-60, 60)
}

# baseline
baseline &lt;- sim_lm(n = 10, effect = 2, error = 5)
baseline_ss_res &lt;- plot_ss_res(baseline)
dir.create(&quot;../figure/ch01/&quot;, recursive = TRUE, showWarnings = FALSE)
ggsave(&quot;../figure/ch01/baseline_ss_res.png&quot;, baseline_ss_res, width = 5, height = 6)
baseline_ss_exp &lt;- plot_ss_exp(baseline)
ggsave(&quot;../figure/ch01/baseline_ss_exp.png&quot;, baseline_ss_exp, width = 5, height = 6)
plot_grid(baseline_ss_res, baseline_ss_exp)</code></pre>
<p><img src="figure/ch01.Rmd/inuition-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>baseline$fstat[1]</code></pre>
<pre><code>[1] 280.3557</code></pre>
<pre class="r"><code># Increased error
more_error &lt;- sim_lm(n = 10, effect = 2, error = 10)
more_error_ss_res &lt;- plot_ss_res(more_error)
dir.create(&quot;../figure/ch01/&quot;, recursive = TRUE, showWarnings = FALSE)
ggsave(&quot;../figure/ch01/more_error_ss_res.png&quot;, more_error_ss_res, width = 5, height = 6)
more_error_ss_exp &lt;- plot_ss_exp(more_error)
ggsave(&quot;../figure/ch01/more_error_ss_exp.png&quot;, more_error_ss_exp, width = 5, height = 6)
plot_grid(more_error_ss_res, more_error_ss_exp)</code></pre>
<p><img src="figure/ch01.Rmd/inuition-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>more_error$fstat[1]</code></pre>
<pre><code>[1] 72.89125</code></pre>
<pre class="r"><code># Decreased signal
less_signal &lt;- sim_lm(n = 10, effect = 1, error = 5)
less_signal_ss_res &lt;- plot_ss_res(less_signal)
dir.create(&quot;../figure/ch01/&quot;, recursive = TRUE, showWarnings = FALSE)
ggsave(&quot;../figure/ch01/less_signal_ss_res.png&quot;, less_signal_ss_res, width = 5, height = 6)
less_signal_ss_exp &lt;- plot_ss_exp(less_signal)
ggsave(&quot;../figure/ch01/less_signal_ss_exp.png&quot;, less_signal_ss_exp, width = 5, height = 6)
plot_grid(less_signal_ss_res, less_signal_ss_exp)</code></pre>
<p><img src="figure/ch01.Rmd/inuition-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>less_signal$fstat[1]</code></pre>
<pre><code>[1] 72.89125</code></pre>
</div>
<div id="standard-linear-model-for-differential-expression-of-one-gene" class="section level2">
<h2>Standard linear model for differential expression of one gene</h2>
<p>Run lm on one gene.</p>
</div>
<div id="standard-linear-model-for-differential-expression-of-many-genes" class="section level2">
<h2>Standard linear model for differential expression of many genes</h2>
<p>Use for loop to run lm on many genes (exact number of genes will be determined by 5 sec time limit for DataCamp exercises).</p>
</div>
<div id="visualize-results-of-standard-differential-expression-analysis" class="section level2">
<h2>Visualize results of standard differential expression analysis</h2>
<p>Use geom_point to plot effect size versus variance and color by statistical significance.</p>
</div>
<div id="limma-linear-models-for-functional-genomics-video" class="section level2">
<h2>limma: linear models for functional genomics (Video)</h2>
<p>Explain the advantages of using limma for functional genomics experiments over traditional linear models: 1) share information across genes to reduce uncertainty of variance, 2) flexible system for specifying contrasts of interest, 3) integrated into standard differential expression workflow. Introduce a limma workflow diagram to orient the individual steps in the following chapters within the larger analysis.</p>
</div>
<div id="limma-for-differential-expression-of-many-genes" class="section level2">
<h2>limma for differential expression of many genes</h2>
<p>Run pre-written limma code to easily test thousands of genes.</p>
</div>
<div id="visualize-results-of-limma-differential-expression-analysis" class="section level2">
<h2>Visualize results of limma differential expression analysis</h2>
<p>Use geom_point to plot effect size versus variance and color by statistical significance.</p>
</div>
<div id="the-benefit-of-sharing-information-across-genes" class="section level2">
<h2>The benefit of sharing information across genes</h2>
<p>Visualize the gene expression pattern of a gene called DE by a traditional linear model but not by limma.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X Yosemite 10.10.5

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cowplot_0.9.2 broom_0.4.2   ggplot2_2.2.1 bindrcpp_0.2  stringr_1.2.0
[6] tidyr_0.6.3   dplyr_0.7.2  

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14     git2r_0.21.0     plyr_1.8.4       bindr_0.1       
 [5] tools_3.3.2      digest_0.6.13    evaluate_0.10.1  tibble_1.3.3    
 [9] gtable_0.2.0     nlme_3.1-128     lattice_0.20-34  pkgconfig_2.0.1 
[13] rlang_0.1.6      psych_1.7.5      yaml_2.1.16      parallel_3.3.2  
[17] knitr_1.18       rprojroot_1.3-2  grid_3.3.2       glue_1.1.1      
[21] R6_2.2.2         foreign_0.8-67   rmarkdown_1.8    reshape2_1.4.2  
[25] magrittr_1.5     backports_1.1.2  scales_0.4.1     htmltools_0.3.6 
[29] assertthat_0.2.0 mnormt_1.5-5     colorspace_1.3-2 labeling_0.3    
[33] stringi_1.1.5    lazyeval_0.2.1   munsell_0.4.3   </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
