<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="John Blischak" />

<meta name="date" content="2018-02-10" />

<title>Introduction to differential expression analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DE analysis with limma</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/dc-bioc-limma">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Introduction to differential expression analysis</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2018-02-10</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-05-23</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/jdblischak/dc-bioc-limma/tree/3e36241704d031a2f9ac82661fdbb77fa7a490fd" target="_blank">3e36241</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    figure/.DS_Store
    Ignored:    figure/ch02/.DS_Store

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/analysis/ch01.Rmd" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
<td style="text-align:left;">
Update to 1.0
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/ch01.html" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
<td style="text-align:left;">
Update to 1.0
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/3144b03e8b47623242cfb24c814c5feffca617ec/docs/ch01.html" target="_blank">3144b03</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-18
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/d4e2da523a9d938608d81707a32fc6a5123066fc/analysis/ch01.Rmd" target="_blank">d4e2da5</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-18
</td>
<td style="text-align:left;">
Add code comments and save plots for slides for ch01.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/645a9dcf1bb9ffcdd4b6a77f2b2ed05f61cf0923/analysis/ch01.Rmd" target="_blank">645a9dc</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
<td style="text-align:left;">
Minor edit.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/201bbf1704ee736496d1cf27791d7069c070c1af/docs/ch01.html" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3571e144da693159351ef73af2893db61977a237/analysis/ch01.Rmd" target="_blank">3571e14</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
<td style="text-align:left;">
Save intermediate data files.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/0c36646f70b7886abbd03da104363af1e38a2f38/analysis/ch01.Rmd" target="_blank">0c36646</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
<td style="text-align:left;">
Continue working on ch01 exercises.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/0536832e58a47905e211708bcda9472a50622305/analysis/ch01.Rmd" target="_blank">0536832</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
<td style="text-align:left;">
Start using new simulation for ch01.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/543057bf600b43ebd88b2754e50c11349ea6200a/docs/ch01.html" target="_blank">543057b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-23
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/f3df390eb4ed78f0f86397d99842aabc0a2eb562/analysis/ch01.Rmd" target="_blank">f3df390</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-23
</td>
<td style="text-align:left;">
Work on linear models slides.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/6209669b434b5aacc26962044a6402ec0fe85e33/docs/ch01.html" target="_blank">6209669</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/13eba3eca2c717870fa4806a5f948af2e155a74d/analysis/ch01.Rmd" target="_blank">13eba3e</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
<td style="text-align:left;">
Document simulation functions.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/0d41acee7a92f8645a5535c62141d95c2470a41e/analysis/ch01.Rmd" target="_blank">0d41ace</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
<td style="text-align:left;">
Improve the plots’ readability and cross-comparability.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/5fff9c0f01f689d84074a9a52d0293c78f7fcacf/analysis/ch01.Rmd" target="_blank">5fff9c0</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
<td style="text-align:left;">
Use explained versus residual sum of squares since it is more straightforward.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b62ae8e60dd7964c9a75ad761840cedeff33a638/analysis/ch01.Rmd" target="_blank">b62ae8e</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
<td style="text-align:left;">
Work on plots to build intution for linear regression.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/aa45d236e3fcc70e3d45c2c0f2f164d20a714ec7/analysis/ch01.Rmd" target="_blank">aa45d23</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
<td style="text-align:left;">
Refactor code to create a synthetic data set for ch 01.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f3b1947fdf6be8500bb210223cf0afc93f992b5/analysis/ch01.Rmd" target="_blank">7f3b194</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-12
</td>
<td style="text-align:left;">
Some ch01 exercises.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/16389719fce474ac98a1f51b12ea296da171e2c4/analysis/ch01.Rmd" target="_blank">1638971</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-12
</td>
<td style="text-align:left;">
Spell check.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/804cd26d41a62cd1cb3716f4c363c0ff44d869c0/analysis/ch01.Rmd" target="_blank">804cd26</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-10
</td>
<td style="text-align:left;">
Start transcript for ch01 video on linear models.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/221239f35e6a05d0c7e3da55ddee2e49da2b562d/analysis/ch01.Rmd" target="_blank">221239f</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-10
</td>
<td style="text-align:left;">
Add scaffolding for ch01 from index.
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<p>Cover the basics of the biology and the statistics.</p>
<pre class="r"><code>gexp &lt;- readRDS(&quot;../data/ch01.rds&quot;)
stats &lt;- readRDS(&quot;../data/ch01-stats.rds&quot;)
de &lt;- readRDS(&quot;../data/de.rds&quot;)
single_gene &lt;- readRDS(&quot;../data/single_gene.rds&quot;)
compare &lt;- readRDS(&quot;../data/compare.rds&quot;)</code></pre>
<div id="what-is-differential-expression-video" class="section level2">
<h2>What is differential expression? (Video)</h2>
<p>Describe a simple model of gene regulation (transcription factors, DNA, RNA, protein) and the functional genomics technologies used to measure the various features. Focus on the type of data they return (continuous vs. discrete).</p>
</div>
<div id="functional-genomics-technologies-multiplechoiceexercise" class="section level2">
<h2>Functional genomics technologies (MultipleChoiceExercise)</h2>
<p>Choose which is true. The options will be various combinations of genomics technologies, what they measure, and the type of data they return. For example, “Microarrays measure RNA levels and return continuous data. ChIP-seq measures transcription factor binding to DNA and returns discrete data.” is true.</p>
</div>
<div id="visualize-one-differentially-expressed-gene" class="section level2">
<h2>Visualize one differentially expressed gene</h2>
<p>Visualize one differentially expressed gene with boxplots (geom_boxplot).</p>
<pre class="r"><code>library(&quot;ggplot2&quot;)

print(single_gene)</code></pre>
<pre><code>  group      gene
1   con  8.555641
2   con  7.977472
3   con 10.388985
4 treat 13.629643
5 treat 15.940644
6 treat 13.341958</code></pre>
<pre class="r"><code># Create boxplots of gene expression for both groups
ggplot(single_gene, aes(x = group, y = gene)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-1-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-1-1.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-1-1.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="visualize-many-differentially-expressed-genes" class="section level2">
<h2>Visualize many differentially expressed genes</h2>
<p>Visualize many differentially expressed genes with a heatmap (stats::heatmap).</p>
<pre class="r"><code># Check the dimensions
dim(gexp)</code></pre>
<pre><code>[1] 100   6</code></pre>
<pre class="r"><code># View the first few rows
head(gexp)</code></pre>
<pre><code>            con1      con2      con3    treat1    treat2   treat3
gene01 11.034305  7.693824 15.220120  5.539417  7.662184 13.37325
gene02  6.681872  8.144218 14.673822 14.551493 12.759165 10.32855
gene03 11.550606 15.551538 10.857151 13.525733 11.723725 12.09515
gene04 14.316391  8.371045 14.139368  7.759454 13.570094 16.37769
gene05  8.845249  7.663989  9.342168 10.534286 13.306553  9.02550
gene06  2.684912 10.244040  4.705581  8.920392  9.902485 10.62007</code></pre>
<pre class="r"><code># Plot a heatmap
heatmap(gexp)</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-2-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-2-1.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-2-1.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/543057bf600b43ebd88b2754e50c11349ea6200a/docs/figure/ch01.Rmd/unnamed-chunk-2-1.png" target="_blank">543057b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/6209669b434b5aacc26962044a6402ec0fe85e33/docs/figure/ch01.Rmd/unnamed-chunk-2-1.png" target="_blank">6209669</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="what-is-a-linear-model-video" class="section level2">
<h2>What is a linear model? (Video)</h2>
<p>Review the basics of multiple linear regression (dependent vs independent variables, continuous vs discrete covariates, intercept term, error term). Demonstrate how to specify equations in R (e.g. <code>lm</code> and <code>model.matrix</code>).</p>
<pre class="r"><code>library(&quot;dplyr&quot;)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)
df_vis &lt;- data.frame(status = rep(c(&quot;con&quot;, &quot;treat&quot;), each = 50)) %&gt;%
  mutate(gene_de = c(rpois(n() / 2, lambda = 12), rpois(n() / 2, lambda = 30)),
         gene_var =c(rpois(n() / 2, lambda = 15) + rnorm(n() / 2, sd = 10),
                     rpois(n() / 2, lambda = 25) + rnorm(n() / 2, sd = 10)))
ggplot(df_vis, aes(x = status, y = gene_de)) +
  geom_boxplot() +
  theme_classic(base_size = 16) +
  ylim(0, 40) +
  labs(x = &quot;Treatment status&quot;, y = &quot;Gene expression level&quot;)</code></pre>
<pre><code>Warning: Removed 1 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-3-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-3-1.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-3-1.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/543057bf600b43ebd88b2754e50c11349ea6200a/docs/figure/ch01.Rmd/unnamed-chunk-3-1.png" target="_blank">543057b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/6209669b434b5aacc26962044a6402ec0fe85e33/docs/figure/ch01.Rmd/unnamed-chunk-3-1.png" target="_blank">6209669</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>ggsave(&quot;../figure/ch01/gene-de.png&quot;, width = 5, height = 6)</code></pre>
<pre><code>Warning: Removed 1 rows containing non-finite values (stat_boxplot).</code></pre>
<pre class="r"><code>ggplot(df_vis, aes(x = status, y = gene_var)) +
  geom_boxplot() +
  theme_classic(base_size = 16) +
  ylim(0, 40) +
  labs(x = &quot;Treatment status&quot;, y = &quot;Gene expression level&quot;)</code></pre>
<pre><code>Warning: Removed 8 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-3-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-3-2.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-3-2.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>ggsave(&quot;../figure/ch01/gene-ambiguous.png&quot;, width = 5, height = 6)</code></pre>
<pre><code>Warning: Removed 8 rows containing non-finite values (stat_boxplot).</code></pre>
<p><strong>Transcript:</strong> As you just visualized, differential expression describes the situation in which a gene has a different mean expression level between conditions. While some gene expression patterns are easily diagnosed as differential expression or not from a quick visualization, you also saw some examples that were more ambiguous. Furthermore, you need a method that is more robust than a quick visual inspection and also scales to thousands of genes. For this you will use the tools of statistical inference to determine if the difference in mean expression level is larger than that expected by chance. Specifically, you will use linear models to perform the hypothesis tests. Linear models are an ideal choice for genomics experiments because their flexibility and robustness to assumptions allow you to conveniently model data from various study designs and data sources.</p>
<p>You should have already been introduced to linear models, for example in a DataCamp course such as Correlation and Regression, or in an introductory statistics course. Here I’ll review the terminology we will use in the remainder of the course, how to specify a linear model in R, and the intuition behind linear models.</p>
<div class="figure">
<img src="../figure/ch01/temperature.png" alt="Temperature experiment" />
<p class="caption">Temperature experiment</p>
</div>
<p><span class="math display">\[ Y =  \beta_0 + \beta_1 X_1 + \epsilon \]</span></p>
<p>In this equation of a linear model, Y is the response variable. It must be a continuous variable. In the context of differential expression, it is a relative measure of either RNA or protein expression level for one gene. <span class="math inline">\(X_1\)</span> is an explanatory variable, which can be continuous or discrete, for example, control group versus treatment, or mutant versus wild type. <span class="math inline">\(\beta_1\)</span> quantifies the effect of the explanatory variable on the response variable. Furthermore, we can add additional explanatory variables to the equation for more complicated experimental designs. Lastly, models the random noise in the measurements.</p>
<p>In R, you specify a linear model with the function <code>lm</code>. This uses R’s formula syntax. On the left is the object that contains the response variable, and to the right of the tilde are the objects that contain the explanatory variables.</p>
<pre class="r"><code>lm(y ~ x1)</code></pre>
<p>A second explanatory variable can be added with a plus sign.</p>
<p><span class="math display">\[ Y =  \beta_0 + \beta_1 X_1 + \beta_2 X_2 + \epsilon \]</span></p>
<pre class="r"><code>lm(y ~ x1 + x2)</code></pre>
<p>Building some intuition.</p>
<pre class="r"><code>library(&quot;broom&quot;)
library(&quot;cowplot&quot;)
library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_classic(base_size = 16))

# Simulate a linear regression.
#
# n = sample size
# effect = slope
# error = standard deviation of distribution of residuals
# seed = seed for random number generator
#
# Returns a data.frame with the following columns:
#
# x          Explanatory variable
# y          Response variable
# y_bar      Mean of response variable
# intercept  Intercept of least squares regression line
# slope      Slope of least squares regression line
# y_hat      Fitted values
# fstat      F-statistic
# ss_exp     Explained sum of squares
# ss_res     Residual sum of squares (noise)
sim_lm &lt;- function(n, effect, error, seed = 1) {
  stopifnot(is.numeric(n), n &gt; 0, length(n) == 1)
  stopifnot(is.numeric(effect), length(effect) == 1)
  stopifnot(is.numeric(error), error &gt; 0, length(error) == 1)
  stopifnot(is.numeric(seed), length(seed) == 1)
  
  set.seed(seed)
  x = runif(n, min = -25, max = 25)
  y = x * effect + rnorm(n, sd = error)
  y_bar = mean(y)
  mod &lt;- lm(y ~ x)
  coefs &lt;- coef(mod)
  intercept &lt;- coefs[1]
  slope &lt;- coefs[2]
  y_hat = fitted(mod)
  anova_tidy &lt;- tidy(anova(mod))
  fstat &lt;- anova_tidy$statistic[1]
  ss &lt;- anova_tidy$sumsq
  ss_exp &lt;- ss[1]
  ss_res &lt;- ss[2]
  stopifnot(ss_exp - sum((y_hat - y_bar)^2) &lt; 0.01)
  stopifnot(ss_res - sum(residuals(mod)^2) &lt; 0.01)
  return(data.frame(x, y, y_bar, intercept, slope, y_hat, fstat, ss_exp, ss_res,
                    row.names = 1:n))
}

# Visualize the residual sum of squares
plot_ss_res &lt;- function(d) {
  ggplot(d, aes(x = x, y = y)) +
    geom_point() +
    geom_abline(aes(intercept = intercept, slope = slope)) +
    geom_linerange(aes(ymin = y, ymax = y_hat), color = &quot;red&quot;,
                   linetype = &quot;dashed&quot;) +
    geom_text(aes(x = quantile(x, 0.25), y = quantile(y, 0.75),
                  label = round(ss_res, 2)), color = &quot;red&quot;) +
    labs(title = &quot;Residual sum of squares (noise)&quot;) +
    ylim(-60, 60)
}

# Visualize the explained sum of squares
plot_ss_exp &lt;- function(d) {
  ggplot(d, aes(x = x, y = y)) +
    geom_abline(aes(intercept = intercept, slope = slope)) +
    geom_hline(aes(yintercept = y_bar)) +
    geom_linerange(aes(ymin = y_hat, ymax = y_bar), color = &quot;blue&quot;,
                   linetype = &quot;dashed&quot;) +
    geom_text(aes(x = quantile(x, 0.25), y = quantile(y, 0.75),
                  label = round(ss_exp, 2)), color = &quot;blue&quot;) +
    labs(title = &quot;Explained sum of squares&quot;) +
    ylim(-60, 60)
}

# baseline
baseline &lt;- sim_lm(n = 10, effect = 2, error = 5)
baseline_ss_res &lt;- plot_ss_res(baseline)
dir.create(&quot;../figure/ch01/&quot;, recursive = TRUE, showWarnings = FALSE)
ggsave(&quot;../figure/ch01/baseline_ss_res.png&quot;, baseline_ss_res, width = 5, height = 6)
baseline_ss_exp &lt;- plot_ss_exp(baseline)
ggsave(&quot;../figure/ch01/baseline_ss_exp.png&quot;, baseline_ss_exp, width = 5, height = 6)
plot_grid(baseline_ss_res, baseline_ss_exp)</code></pre>
<p><img src="figure/ch01.Rmd/inuition-1.png" width="960" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of inuition-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/543057bf600b43ebd88b2754e50c11349ea6200a/docs/figure/ch01.Rmd/inuition-1.png" target="_blank">543057b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/6209669b434b5aacc26962044a6402ec0fe85e33/docs/figure/ch01.Rmd/inuition-1.png" target="_blank">6209669</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>baseline$fstat[1]</code></pre>
<pre><code>[1] 280.3557</code></pre>
<pre class="r"><code># Increased error
more_error &lt;- sim_lm(n = 10, effect = 2, error = 10)
more_error_ss_res &lt;- plot_ss_res(more_error)
dir.create(&quot;../figure/ch01/&quot;, recursive = TRUE, showWarnings = FALSE)
ggsave(&quot;../figure/ch01/more_error_ss_res.png&quot;, more_error_ss_res, width = 5, height = 6)
more_error_ss_exp &lt;- plot_ss_exp(more_error)
ggsave(&quot;../figure/ch01/more_error_ss_exp.png&quot;, more_error_ss_exp, width = 5, height = 6)
plot_grid(more_error_ss_res, more_error_ss_exp)</code></pre>
<p><img src="figure/ch01.Rmd/inuition-2.png" width="960" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of inuition-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/543057bf600b43ebd88b2754e50c11349ea6200a/docs/figure/ch01.Rmd/inuition-2.png" target="_blank">543057b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/6209669b434b5aacc26962044a6402ec0fe85e33/docs/figure/ch01.Rmd/inuition-2.png" target="_blank">6209669</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>more_error$fstat[1]</code></pre>
<pre><code>[1] 72.89125</code></pre>
<pre class="r"><code># Decreased signal
less_signal &lt;- sim_lm(n = 10, effect = 1, error = 5)
less_signal_ss_res &lt;- plot_ss_res(less_signal)
dir.create(&quot;../figure/ch01/&quot;, recursive = TRUE, showWarnings = FALSE)
ggsave(&quot;../figure/ch01/less_signal_ss_res.png&quot;, less_signal_ss_res, width = 5, height = 6)
less_signal_ss_exp &lt;- plot_ss_exp(less_signal)
ggsave(&quot;../figure/ch01/less_signal_ss_exp.png&quot;, less_signal_ss_exp, width = 5, height = 6)
plot_grid(less_signal_ss_res, less_signal_ss_exp)</code></pre>
<p><img src="figure/ch01.Rmd/inuition-3.png" width="960" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of inuition-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/543057bf600b43ebd88b2754e50c11349ea6200a/docs/figure/ch01.Rmd/inuition-3.png" target="_blank">543057b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/6209669b434b5aacc26962044a6402ec0fe85e33/docs/figure/ch01.Rmd/inuition-3.png" target="_blank">6209669</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-02-16
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>less_signal$fstat[1]</code></pre>
<pre><code>[1] 72.89125</code></pre>
</div>
<div id="standard-linear-model-for-differential-expression-of-one-gene" class="section level2">
<h2>Standard linear model for differential expression of one gene</h2>
<p>Run lm on one gene.</p>
<pre class="r"><code>head(single_gene)</code></pre>
<pre><code>  group      gene
1   con  8.555641
2   con  7.977472
3   con 10.388985
4 treat 13.629643
5 treat 15.940644
6 treat 13.341958</code></pre>
<pre class="r"><code># Fit the linear model
mod &lt;- lm(gene ~ group, data = single_gene)

# Summarize the results
result &lt;- summary(mod)
result$coefficients</code></pre>
<pre><code>            Estimate Std. Error   t value     Pr(&gt;|t|)
(Intercept) 8.974033  0.7761637 11.562035 0.0003196394
grouptreat  5.330049  1.0976613  4.855823 0.0083043428</code></pre>
</div>
<div id="standard-linear-model-for-differential-expression-of-many-genes" class="section level2">
<h2>Standard linear model for differential expression of many genes</h2>
<p>Use for loop to run lm on many genes (exact number of genes will be determined by 5 sec time limit for DataCamp exercises).</p>
<pre class="r"><code>group &lt;- rep(c(&quot;con&quot;, &quot;treat&quot;), each = ncol(gexp) / 2)</code></pre>
<pre class="r"><code>dim(gexp)</code></pre>
<pre><code>[1] 100   6</code></pre>
<pre class="r"><code>print(group)</code></pre>
<pre><code>[1] &quot;con&quot;   &quot;con&quot;   &quot;con&quot;   &quot;treat&quot; &quot;treat&quot; &quot;treat&quot;</code></pre>
<pre class="r"><code># Initialize a vector to save the p-value from testing each gene
p &lt;- numeric(length = nrow(gexp))

# Use a for loop to test each gene
for (i in 1:length(p)) {
  # Fit the model
  mod &lt;- lm(gexp[i, ] ~ group)
  # Extract the p-value
  result &lt;- summary(mod)
  p[i] &lt;- result$coefficients[2, 4]
}

# Visualize p-value distribution with a histogram
hist(p)</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-8-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-8-1.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-8-1.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="visualize-results-of-standard-differential-expression-analysis" class="section level2">
<h2>Visualize results of standard differential expression analysis</h2>
<p>Use geom_point to plot effect size versus variance and color by statistical significance.</p>
<pre class="r"><code>head(de)</code></pre>
<pre><code>  effect_size  std_dev    lm limma
1  -2.4577980 3.750555 FALSE FALSE
2   2.7130980 3.353835 FALSE FALSE
3  -0.2048963 1.715542 FALSE FALSE
4   0.2934781 3.511502 FALSE FALSE
5   2.3383110 1.955378 FALSE FALSE
6   3.9361382 3.326517 FALSE FALSE</code></pre>
<pre class="r"><code># Plot effect size (y-axis) vs. standard deviation (x-axis)
ggplot(de, aes(x = std_dev, y = effect_size, color = lm)) +
  geom_point()</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-9-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-9-1.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-9-1.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="limma-linear-models-for-functional-genomics-video" class="section level2">
<h2>limma: linear models for functional genomics (Video)</h2>
<p>Explain the advantages of using limma for functional genomics experiments over traditional linear models: 1) share information across genes to reduce uncertainty of variance, 2) flexible system for specifying contrasts of interest, 3) integrated into standard differential expression workflow. Introduce a limma workflow diagram to orient the individual steps in the following chapters within the larger analysis.</p>
</div>
<div id="limma-for-differential-expression-of-many-genes" class="section level2">
<h2>limma for differential expression of many genes</h2>
<p>Run pre-written limma code to easily test thousands of genes.</p>
<pre class="r"><code>library(&quot;limma&quot;)
print(group)</code></pre>
<pre><code>[1] &quot;con&quot;   &quot;con&quot;   &quot;con&quot;   &quot;treat&quot; &quot;treat&quot; &quot;treat&quot;</code></pre>
<pre class="r"><code># Setup the study design matrix
design &lt;- model.matrix(~group)
colnames(design) &lt;- c(&quot;Intercept&quot;, &quot;group&quot;)

# Fit the model
fit &lt;- lmFit(gexp, design)
fit &lt;- eBayes(fit)

# Summarize results
results &lt;- decideTests(fit[, &quot;group&quot;])
summary(results)</code></pre>
<pre><code>   group
-1    15
0     71
1     14</code></pre>
</div>
<div id="visualize-results-of-limma-differential-expression-analysis" class="section level2">
<h2>Visualize results of limma differential expression analysis</h2>
<p>Use geom_point to plot effect size versus variance and color by statistical significance.</p>
<pre class="r"><code>head(de)</code></pre>
<pre><code>  effect_size  std_dev    lm limma
1  -2.4577980 3.750555 FALSE FALSE
2   2.7130980 3.353835 FALSE FALSE
3  -0.2048963 1.715542 FALSE FALSE
4   0.2934781 3.511502 FALSE FALSE
5   2.3383110 1.955378 FALSE FALSE
6   3.9361382 3.326517 FALSE FALSE</code></pre>
<pre class="r"><code># View the number of discrepancies
table(de$lm, de$limma)</code></pre>
<pre><code>       
        FALSE TRUE
  FALSE    68    7
  TRUE      3   22</code></pre>
<pre class="r"><code># Plot effect size (y-axis) vs. standard deviation (x-axis)
ggplot(de, aes(x = std_dev, y = effect_size, color = limma)) +
  geom_point()</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-11-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-11-1.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-11-1.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="the-benefit-of-sharing-information-across-genes" class="section level2">
<h2>The benefit of sharing information across genes</h2>
<p>Visualize the gene expression pattern of a gene called DE by a traditional linear model but not by limma.</p>
<pre class="r"><code>head(compare)</code></pre>
<pre><code>        type group      gene
1    neither   con  6.681872
2       both   con  8.555641
3    lm-only   con  9.959914
4 limma-only   con 11.391149
5    neither   con  8.144218
6       both   con  7.977472</code></pre>
<pre class="r"><code># Plot gene expression (gene; y-axis) vs. group (x-axis)
ggplot(compare, aes(x = group, y = gene)) +
  geom_boxplot() +
  facet_wrap(~type, nrow = 1)</code></pre>
<p><img src="figure/ch01.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-12-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/figure/ch01.Rmd/unnamed-chunk-12-1.png" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/201bbf1704ee736496d1cf27791d7069c070c1af/docs/figure/ch01.Rmd/unnamed-chunk-12-1.png" target="_blank">201bbf1</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-04-06
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X Yosemite 10.10.5

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] limma_3.30.13 cowplot_0.9.2 broom_0.4.4   bindrcpp_0.2  dplyr_0.7.4  
[6] ggplot2_2.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14      pillar_1.2.2      git2r_0.21.0     
 [4] plyr_1.8.4        workflowr_1.0.1   bindr_0.1.1      
 [7] R.methodsS3_1.7.1 R.utils_2.6.0     tools_3.3.3      
[10] digest_0.6.13     evaluate_0.10.1   tibble_1.4.2     
[13] gtable_0.2.0      nlme_3.1-131      lattice_0.20-35  
[16] pkgconfig_2.0.1   rlang_0.2.0       psych_1.8.3.3    
[19] parallel_3.3.3    yaml_2.1.16       stringr_1.3.0    
[22] knitr_1.20        rprojroot_1.3-2   grid_3.3.3       
[25] glue_1.2.0        R6_2.2.2          foreign_0.8-69   
[28] rmarkdown_1.9.12  reshape2_1.4.3    tidyr_0.7.2      
[31] purrr_0.2.4       magrittr_1.5      whisker_0.3-2    
[34] backports_1.1.2   scales_0.5.0      htmltools_0.3.6  
[37] mnormt_1.5-5      assertthat_0.2.0  colorspace_1.3-2 
[40] labeling_0.3      stringi_1.1.7     lazyeval_0.2.1   
[43] munsell_0.4.3     R.oo_1.22.0      </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.0.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
