<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="John Blischak" />


<title>Differential expression analysis for studies with more than 2 groups</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DE analysis with limma</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/dc-bioc-limma">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential expression analysis for studies with more than 2 groups</h1>
<h4 class="author"><em>John Blischak</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-05-23</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/jdblischak/dc-bioc-limma/tree/cc6353a03cb2d5c5c068d9f3e25f1cc3a2adbf0b" target="_blank">cc6353a</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    figure/.DS_Store
    Ignored:    figure/ch02/.DS_Store

Untracked files:
    Untracked:  _workflowr.yml

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/ch01.Rmd
    Modified:   analysis/ch02.Rmd
    Modified:   analysis/ch03.Rmd
    Modified:   analysis/ch04.Rmd
    Deleted:    analysis/chunks.R
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/ch03.html" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/9a54d7c1a7281865b706d32a8bb01d917205f7cc/analysis/ch03.Rmd" target="_blank">9a54d7c</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
<td style="text-align:left;">
Add code for Ch 03
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/3bcf1680c57f2c944ef2f5e02a59facd86f4f565/docs/ch03.html" target="_blank">3bcf168</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/6395f07170f487947a75ccfd7d4bc0b8561148b3/analysis/ch03.Rmd" target="_blank">6395f07</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
<td style="text-align:left;">
Start organizing Ch 03
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<p>Analyze RNA-seq data from Hoban et al., 2016, which measured gene expression in the prefrontal cortex of 3 groups of mice: conventional (con), germ-free (gf), and germ-free colonized with normal microbiota (exgf).</p>
<pre class="r"><code>library(&quot;Biobase&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;limma&quot;)
library(&quot;stringr&quot;)
dge &lt;- readRDS(&quot;../data/ch03.rds&quot;)</code></pre>
<div id="studies-with-more-than-2-groups-video" class="section level2">
<h2>Studies with more than 2 groups (Video)</h2>
<p>Describe the scientific question, the experimental design, and the data collected for the 3-group study. Review RNA-seq technique and the need to standardize by library size.</p>
</div>
<div id="calculate-the-library-size" class="section level2">
<h2>Calculate the library size</h2>
<p>Use <code>colSums</code> to calculate the library sizes and run <code>summary</code> to appreciate the variability.</p>
<pre class="r"><code>lib_size &lt;- colSums(dge$counts)
summary(lib_size)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
14710000 24330000 26530000 26030000 29460000 33080000 </code></pre>
</div>
<div id="calculate-counts-per-million" class="section level2">
<h2>Calculate counts per million</h2>
<p>Use <code>edgeR::calcNormFactors</code> and <code>edgeR::cpm</code> to calculate TMM-normalized counts per million.</p>
<pre class="r"><code>dge &lt;- calcNormFactors(dge)
dge_cpm &lt;- cpm(dge, log = TRUE)</code></pre>
</div>
<div id="create-density-plots-of-counts-versus-counts-per-million" class="section level2">
<h2>Create density plots of counts versus counts per million</h2>
<p>Use <code>limma::plotDensities</code> to visualize the distribution of counts and the distribution of counts per million.</p>
<pre class="r"><code>plotDensities(dge_cpm, legend = FALSE)</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-3-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-3-1.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="filtering-genes-and-samples-video" class="section level2">
<h2>Filtering genes and samples (Video)</h2>
<p>Describe the process of filtering lowly expressed genes (show density plots from previous exercise to demonstrate bimodal distribution) and identifying outlier samples with PCA. Discuss the nuances of removing outlier samples.</p>
</div>
<div id="remove-lowly-expressed-genes" class="section level2">
<h2>Remove lowly expressed genes</h2>
<p>Remove any gene that does not have at least a log2 cpm &gt; 0 in at least 4 samples (n = 4 per group). Use <code>limma::plotDensities</code> to confirm the distribution of counts per million is unimodal post-filtering.</p>
<pre class="r"><code>dge_filt &lt;- dge[rowSums(dge_cpm &gt; 0) &gt;= 4, ]
dge_filt &lt;- calcNormFactors(dge_filt)
dge_filt_cpm &lt;- cpm(dge_filt, log = TRUE)
plotDensities(dge_filt_cpm, legend = FALSE)</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-4-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-4-1.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="use-pca-to-identify-outlier-sample" class="section level2">
<h2>Use PCA to identify outlier sample</h2>
<p>Use <code>prcomp</code> to calculate PCA and then plot PC1 vs. PC2. Remove the exgf sample that clusters with the gf samples.</p>
<pre class="r"><code>pca &lt;- prcomp(t(dge_filt_cpm), scale. = TRUE)
plot(pca)</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-5-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-5-1.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>d &lt;- data.frame(dge_filt$samples, pca$x)
ggplot(d, aes(x = PC1, y = PC2, color = group)) +
  geom_text(aes(label = id))</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-5-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-5-2.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>plotMDS(dge_filt)</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-5-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-5-3.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>dge_final &lt;- dge_filt[, dge_filt$samples$id != &quot;exgf1&quot;]</code></pre>
</div>
<div id="differential-expression-for-rna-seq-with-voom-video" class="section level2">
<h2>Differential expression for RNA-seq with voom (Video)</h2>
<p>Review how to use <code>model.matrix</code> and <code>limma::makeContrasts</code>. Describe how <code>limma::voom</code> corrects for the mean-variance relationship of the count data.</p>
</div>
<div id="create-design-matrix-for-study-with-3-groups" class="section level2">
<h2>Create design matrix for study with 3 groups</h2>
<p>Use <code>model.matrix</code> to create a linear model with three binary variables (group-means parametrization).</p>
<pre class="r"><code>design &lt;- model.matrix(~0 + group, data = dge_final$samples)
colnames(design) &lt;- str_replace(colnames(design), &quot;group&quot;, &quot;&quot;)
colSums(design)</code></pre>
<pre><code> con exgf   gf 
   4    3    4 </code></pre>
</div>
<div id="create-contrasts-matrix-for-study-with-3-groups" class="section level2">
<h2>Create contrasts matrix for study with 3 groups</h2>
<p>Use <code>limma::makeContrasts</code> to test all pairwise comparisions.</p>
<pre class="r"><code>cont_mat &lt;- makeContrasts(gf.con = gf - con,
                          exgf.con = exgf - con,
                          exgf.gf = exgf - gf,
                          levels = design)
cont_mat</code></pre>
<pre><code>      Contrasts
Levels gf.con exgf.con exgf.gf
  con      -1       -1       0
  exgf      0        1       1
  gf        1        0      -1</code></pre>
</div>
<div id="account-for-mean-variance-relationship-with-voom" class="section level2">
<h2>Account for mean-variance relationship with voom</h2>
<p>Use <code>limma::voom</code> to calculate weights for the linear model to account for the mean-variance relationship of the counts. Set <code>plot = TRUE</code> to visualize the mean-variance trend. Explore the EList object returned to confirm that the expression values in <code>E</code> are simply the counts per million.</p>
<pre class="r"><code>dge_final &lt;- calcNormFactors(dge_final)
v &lt;- voom(dge_final, design, plot = TRUE)</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-8-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-8-1.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>plotDensities(v$E, legend = FALSE)</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-8-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-8-2.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>dge_final_cpm &lt;- cpm(dge_final, log = TRUE)
plot(v$E[, 1], dge_final_cpm[, 1])</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-8-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-8-3.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="fit-and-test-model-for-study-with-3-groups" class="section level2">
<h2>Fit and test model for study with 3 groups</h2>
<p>Use <code>limma::lmFit</code>, <code>limma::contrasts.fit</code>, <code>limma::eBayes</code>, and <code>limma::decideTests</code> to fit and test the model. Use <code>limma::vennDiagram</code> to visualize overlap of differentially expressed genes.</p>
<pre class="r"><code>fit &lt;- lmFit(v, design)
fit2 &lt;- contrasts.fit(fit, contrasts = cont_mat)
fit2 &lt;- eBayes(fit2)
results &lt;- decideTests(fit2)
vennDiagram(results)</code></pre>
<p><img src="figure/ch03.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-9-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/77a43dd33fa053ff49b73d253d2e1389d4d5f808/docs/figure/ch03.Rmd/unnamed-chunk-9-1.png" target="_blank">77a43dd</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-12
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X Yosemite 10.10.5

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] stringr_1.3.0       ggplot2_2.2.1       edgeR_3.16.5       
[4] limma_3.30.13       Biobase_2.34.0      BiocGenerics_0.20.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14      knitr_1.20        whisker_0.3-2    
 [4] magrittr_1.5      workflowr_1.0.1   munsell_0.4.3    
 [7] colorspace_1.3-2  lattice_0.20-35   rlang_0.2.0      
[10] plyr_1.8.4        tools_3.3.3       grid_3.3.3       
[13] gtable_0.2.0      R.oo_1.22.0       git2r_0.21.0     
[16] htmltools_0.3.6   lazyeval_0.2.1    yaml_2.1.16      
[19] rprojroot_1.3-2   digest_0.6.13     tibble_1.4.2     
[22] R.utils_2.6.0     evaluate_0.10.1   rmarkdown_1.9.12 
[25] labeling_0.3      stringi_1.1.7     pillar_1.2.2     
[28] scales_0.5.0      backports_1.1.2   R.methodsS3_1.7.1
[31] locfit_1.5-9.1   </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.0.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
