<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="John Blischak" />


<title>Differential expression analysis for studies with a 2x2 factorial design</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DE analysis with limma</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/dc-bioc-limma">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential expression analysis for studies with a 2x2 factorial design</h1>
<h4 class="author"><em>John Blischak</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-06-15</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/jdblischak/dc-bioc-limma/tree/d979358369cc86e81e2e8ae1697ab3ca06891847" target="_blank">d979358</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/counts_per_sample.txt
    Untracked:  code/table-s1.txt

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/d979358369cc86e81e2e8ae1697ab3ca06891847/analysis/ch04.Rmd" target="_blank">d979358</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-06-15
</td>
<td style="text-align:left;">
Use all lowercase variables for dox phenotypes.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/c36122519ba4fa621f2d35809ded167ccb3d4183/analysis/ch04.Rmd" target="_blank">c361225</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-30
</td>
<td style="text-align:left;">
Experiment with normalizing and filtering doxorubicin data.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/9b58c2521a3c4a74cb302637d4ac089ad8573273/docs/ch04.html" target="_blank">9b58c25</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/79a1ea4f672305aab1ce2ba4ab93f8522a8bafb3/docs/ch04.html" target="_blank">79a1ea4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/3e36241704d031a2f9ac82661fdbb77fa7a490fd/analysis/ch04.Rmd" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
<td style="text-align:left;">
Update to 1.0
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/3e36241704d031a2f9ac82661fdbb77fa7a490fd/docs/ch04.html" target="_blank">3e36241</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-05-23
</td>
<td style="text-align:left;">
Update to 1.0
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/ch04.html" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/a2fc9a7de10a2a1bd7c7aa3566c3bb7d47ba372c/analysis/ch04.Rmd" target="_blank">a2fc9a7</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
<td style="text-align:left;">
Minor edits to ch04 solutions
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/ch04.html" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/dc823825da0b6d5f60a9af5a31a2cb47fa735a7e/analysis/ch04.Rmd" target="_blank">dc82382</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
<td style="text-align:left;">
Minor updates to Ch 04 analysis.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/ch04.html" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/5db0175b96175b7b49c77fabff46836abc10a4a5/analysis/ch04.Rmd" target="_blank">5db0175</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
<td style="text-align:left;">
Organize Ch 04
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>Analayze microarray data from Zhang et al., 2012, which measured gene expression in hearts from wild type and Top2b null mice treated with doxorubicin or a control.</p>
<pre class="r"><code>library(&quot;Biobase&quot;)
library(&quot;ggplot2&quot;)
library(&quot;limma&quot;)
eset &lt;- readRDS(&quot;../data/ch04.rds&quot;)</code></pre>
<div id="studies-with-a-2x2-factorial-design-video" class="section level2">
<h2>Studies with a 2x2 factorial design (Video)</h2>
<p>Describe the scientific question, the experimental design, and the data collected for the 2x2 factorial study.</p>
</div>
<div id="expression-level-of-top2b" class="section level2">
<h2>Expression level of Top2b</h2>
<p>Create a boxplot of the expression level of Top2b to confirm the null mice have lower levels of Top2b expression. Use <code>pData</code> and <code>table</code> to access and explore the phenotype measurements to be used in the plot.</p>
<pre class="r"><code>head(fData(eset))</code></pre>
<pre><code>           probe chrom     start       end symbol            ensembl
1  A_55_P2051983 chr18  62753954  62753895 Spink7 ENSMUST00000076194
2   A_52_P169082 chr11  76031955  76032014  Dbil5                   
3 A_30_P01028193 chr14  47982421  47982480                          
4   A_52_P237997  chr3 156834624 156834683  Negr1                   
5   A_51_P414243  chr9 121890800 121890741 C85492 ENSMUST00000084743
6  A_55_P2136348  chr9 120003293 120003352   Ccr8 ENSMUST00000048777
        refseq entrez                                                name
1 NM_001001803 408198 serine peptidase inhibitor, Kazal type 7 (putative)
2    NM_021294  13168                   diazepam binding inhibitor-like 5
3                  NA                                                    
4    NM_177274 320840                         neuronal growth regulator 1
5    NM_153540 215494                           expressed sequence C85492
6    NM_007720  12776                    chemokine (C-C motif) receptor 8</code></pre>
<pre class="r"><code>top2b &lt;- exprs(eset)[fData(eset)$symbol == &quot;Top2b&quot;, ]
head(pData(eset))</code></pre>
<pre><code>                                                  title genotype treatment
wt.pbs.r1         WT Cardiomyocytes 16hr after PBS rep1       wt       pbs
wt.pbs.r2         WT Cardiomyocytes 16hr after PBS rep2       wt       pbs
wt.pbs.r3         WT Cardiomyocytes 16hr after PBS rep3       wt       pbs
wt.dox.r1 WT Cardiomyocytes 16hr after doxorubicin rep1       wt       dox
wt.dox.r2 WT Cardiomyocytes 16hr after doxorubicin rep2       wt       dox
wt.dox.r3 WT Cardiomyocytes 16hr after doxorubicin rep3       wt       dox
          rep
wt.pbs.r1  r1
wt.pbs.r2  r2
wt.pbs.r3  r3
wt.dox.r1  r1
wt.dox.r2  r2
wt.dox.r3  r3</code></pre>
<pre class="r"><code>d &lt;- data.frame(pData(eset), top2b)
ggplot(d, aes(x = treatment, y = top2b, fill = genotype)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-1-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-1-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-1-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-1-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="visualize-gene-expression-distribution-with-boxplots" class="section level2">
<h2>Visualize gene expression distribution with boxplots</h2>
<p>Use boxplots as an alternative to density plots for checking the distribution of gene expression levels. Note pro (easier to see which sample belongs to each distribution) and con (hides any multimodality).</p>
<pre class="r"><code>boxplot(exprs(eset))</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-2-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-2-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-2-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-2-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>boxplot(log(exprs(eset)))</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-2-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-2-2.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>boxplot(log(exprs(eset)), las = 2)</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-2-3.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-2-3.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>plotDensities(log(exprs(eset)), group = pData(eset)$genotype, legend = &quot;topright&quot;)</code></pre>
<p><img src="figure/ch04.Rmd/densities-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>exprs(eset) &lt;- normalizeBetweenArrays(log(exprs(eset)))
plotDensities(exprs(eset), group = pData(eset)$genotype, legend = &quot;topright&quot;)</code></pre>
<p><img src="figure/ch04.Rmd/densities-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>eset &lt;- eset[rowMeans(exprs(eset)) &gt; 0, ]
dim(eset)</code></pre>
<pre><code>Features  Samples 
   28965       12 </code></pre>
<pre class="r"><code>plotDensities(exprs(eset), group = pData(eset)$genotype, legend = &quot;topright&quot;)</code></pre>
<p><img src="figure/ch04.Rmd/densities-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="perform-pca" class="section level2">
<h2>Perform PCA</h2>
<p>Use <code>prcomp</code> to calculate PCs and plot PC2 vs. PC1 to confirm that the samples are separated by genotype and treatment. Note that Top2b mice all cluster with untreated WT samples since they are resistant to the treatment.</p>
<pre class="r"><code>pca &lt;- prcomp(t(exprs(eset)), scale. = TRUE)
plot(pca)</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-3-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>d &lt;- data.frame(pData(eset), pca$x)
ggplot(d, aes(x = PC1, y = PC2, color = genotype, shape = treatment)) +
  geom_point()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-3-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-2.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-2.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-2.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>ggplot(d, aes(x = treatment, y = PC1, fill = genotype)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-3-3.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-3.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-3.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-3.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>ggplot(d, aes(x = treatment, y = PC2, fill = genotype)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-3-4.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-4.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-4.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-4.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="differential-expression-for-2x2-factorial-designs-video" class="section level2">
<h2>Differential expression for 2x2 factorial designs (Video)</h2>
<p>Discuss how to construct a design and contrasts matrix for a 2x2 factorial design. Note how to create a combined factor and how to write the contrast for the interaction term.</p>
</div>
<div id="create-design-matrix-for-2x2-factorial-design" class="section level2">
<h2>Create design matrix for 2x2 factorial design</h2>
<p>Create a combined factor using <code>paste</code>. Use <code>model.matrix</code> to create a linear model with 4 binary variables (group-means parametrization).</p>
<pre class="r"><code>f &lt;- paste(pData(eset)$genotype, pData(eset)$treatment, sep = &quot;.&quot;)
design &lt;- model.matrix(~0 + f)
colnames(design) &lt;- sub(&quot;f&quot;, &quot;&quot;, colnames(design))
colSums(design)</code></pre>
<pre><code>top2b.dox top2b.pbs    wt.dox    wt.pbs 
        3         3         3         3 </code></pre>
</div>
<div id="create-contrasts-matrix-for-2x2-factorial-design" class="section level2">
<h2>Create contrasts matrix for 2x2 factorial design</h2>
<p>Use <code>limma::makeContrasts</code> to test the main effects and the interaction between genotype and treatment.</p>
<pre class="r"><code>cont_mat &lt;- makeContrasts(DoxWT = wt.dox - wt.pbs,
                          DoxTop = top2b.dox - top2b.pbs,
                          Interaction = (top2b.dox - top2b.pbs) -
                                        (wt.dox - wt.pbs),
                          levels = design)
cont_mat</code></pre>
<pre><code>           Contrasts
Levels      DoxWT DoxTop Interaction
  top2b.dox     0      1           1
  top2b.pbs     0     -1          -1
  wt.dox        1      0          -1
  wt.pbs       -1      0           1</code></pre>
</div>
<div id="fit-and-test-2x2-factorial-design" class="section level2">
<h2>Fit and test 2x2 factorial design</h2>
<p>Use <code>limma::lmFit</code>, <code>limma::contrasts.fit</code>, <code>limma::eBayes</code>, and <code>limma::decideTests</code> to fit and test the model. Use <code>limma::vennDiagram</code> to visualize overlap of differentially expressed genes.</p>
<pre class="r"><code>fit &lt;- lmFit(eset, design)
head(fit$coefficients)</code></pre>
<pre><code>   top2b.dox top2b.pbs   wt.dox   wt.pbs
2   1.405056  1.536537 1.280498 1.172957
5   5.479908  5.486018 5.628112 5.656115
10  1.987281  1.913208 2.206463 2.162715
11  5.688783  5.789735 5.314792 5.283211
14  5.384812  5.406607 5.181832 5.499237
15  1.876791  1.809405 1.765531 1.833199</code></pre>
<pre class="r"><code>fit2 &lt;- contrasts.fit(fit, contrasts = cont_mat)
head(fit2$coefficients)</code></pre>
<pre><code>    Contrasts
           DoxWT       DoxTop Interaction
  2   0.10754056 -0.131481235 -0.23902179
  5  -0.02800370 -0.006109353  0.02189435
  10  0.04374777  0.074073731  0.03032596
  11  0.03158122 -0.100951885 -0.13253311
  14 -0.31740484 -0.021795427  0.29560942
  15 -0.06766808  0.067386526  0.13505461</code></pre>
<pre class="r"><code>fit2 &lt;- eBayes(fit2)
results &lt;- decideTests(fit2)
summary(results)</code></pre>
<pre><code>       DoxWT DoxTop Interaction
Down    4235      0        1427
NotSig 20857  28965       25687
Up      3873      0        1851</code></pre>
<pre class="r"><code>vennDiagram(results)</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-6-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-6-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-6-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-6-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="enrichment-analysis-video" class="section level2">
<h2>Enrichment analysis (Video)</h2>
<p>Describe common approaches for testing for enrichment of gene sets. Discuss Gene Onotlogy categories and KEGG pathways. Introduce Fisher’s Exact Test. Stress the importance of choosing the correct background set of genes and caution against over-interpretation (especially for directional effect). Mention at the end that there are more sophisticated algorithms such as <code>limma::camera</code>, <code>limma::roast</code>, and <code>goseq::goseq</code>.</p>
</div>
<div id="enrichment-of-kegg-pathways" class="section level2">
<h2>Enrichment of KEGG pathways</h2>
<p>Use <code>limma::kegga</code> and <code>limma::topKEGG</code> to test for enrichment of differentially expressed genes in KEGG pathways.</p>
<pre class="r"><code>entrez &lt;- fit2$genes$entrez
enrich_kegg_wt &lt;- kegga(fit2, coef = &quot;DoxWT&quot;,
                        geneid = entrez, species = &quot;Mm&quot;)
topKEGG(enrich_kegg_wt)</code></pre>
<pre><code>                                                             Pathway   N
path:mmu05322                           Systemic lupus erythematosus  76
path:mmu05034                                             Alcoholism 130
path:mmu05330                                    Allograft rejection  26
path:mmu05412 Arrhythmogenic right ventricular cardiomyopathy (ARVC)  52
path:mmu03008                      Ribosome biogenesis in eukaryotes  71
path:mmu05332                              Graft-versus-host disease  28
path:mmu05203                                   Viral carcinogenesis 172
path:mmu04940                               Type I diabetes mellitus  31
path:mmu04261                 Adrenergic signaling in cardiomyocytes 113
path:mmu04137                                     Mitophagy - animal  62
path:mmu04020                              Calcium signaling pathway 125
path:mmu04217                                            Necroptosis 134
path:mmu05320                             Autoimmune thyroid disease  34
path:mmu04974                       Protein digestion and absorption  47
path:mmu04022                             cGMP-PKG signaling pathway 131
path:mmu05168                               Herpes simplex infection 159
path:mmu04919                      Thyroid hormone signaling pathway 104
path:mmu04612                    Antigen processing and presentation  53
path:mmu05167      Kaposi&#39;s sarcoma-associated herpesvirus infection 158
path:mmu04115                                  p53 signaling pathway  60
              Up Down         P.Up       P.Down
path:mmu05322 37    1 2.587115e-11 9.999998e-01
path:mmu05034 46   17 4.415300e-08 9.586496e-01
path:mmu05330 16    0 1.786229e-07 1.000000e+00
path:mmu05412  1   26 9.998834e-01 2.255483e-07
path:mmu03008 29    4 4.415352e-07 9.995765e-01
path:mmu05332 16    0 7.420483e-07 1.000000e+00
path:mmu05203 51   22 4.226308e-06 9.811657e-01
path:mmu04940 16    0 4.532716e-06 1.000000e+00
path:mmu04261 13   40 9.282700e-01 1.261940e-05
path:mmu04137 24   12 1.285527e-05 4.727691e-01
path:mmu04020 10   43 9.972075e-01 1.381343e-05
path:mmu04217 41   21 1.568944e-05 8.220904e-01
path:mmu05320 16    1 2.057586e-05 9.990078e-01
path:mmu04974  2   21 9.972219e-01 2.969888e-05
path:mmu04022  9   43 9.994705e-01 5.002701e-05
path:mmu05168 45   21 5.406625e-05 9.681379e-01
path:mmu04919 12   36 9.195045e-01 5.790107e-05
path:mmu04612 20    6 1.015832e-04 9.419283e-01
path:mmu05167 43   26 2.058751e-04 7.649225e-01
path:mmu04115 21    6 2.376781e-04 9.751968e-01</code></pre>
<pre class="r"><code>enrich_kegg_inter &lt;- kegga(fit2, coef = &quot;Interaction&quot;,
                        geneid = entrez, species = &quot;Mm&quot;)
topKEGG(enrich_kegg_inter)</code></pre>
<pre><code>                                                                 Pathway
path:mmu05322                               Systemic lupus erythematosus
path:mmu05034                                                 Alcoholism
path:mmu03008                          Ribosome biogenesis in eukaryotes
path:mmu05412     Arrhythmogenic right ventricular cardiomyopathy (ARVC)
path:mmu04115                                      p53 signaling pathway
path:mmu04217                                                Necroptosis
path:mmu05414                               Dilated cardiomyopathy (DCM)
path:mmu05203                                       Viral carcinogenesis
path:mmu05202                    Transcriptional misregulation in cancer
path:mmu05410                          Hypertrophic cardiomyopathy (HCM)
path:mmu00534 Glycosaminoglycan biosynthesis - heparan sulfate / heparin
path:mmu03013                                              RNA transport
path:mmu05142                  Chagas disease (American trypanosomiasis)
path:mmu04012                                     ErbB signaling pathway
path:mmu04137                                         Mitophagy - animal
path:mmu05152                                               Tuberculosis
path:mmu03040                                                Spliceosome
path:mmu04020                                  Calcium signaling pathway
path:mmu05205                                    Proteoglycans in cancer
path:mmu04015                                     Rap1 signaling pathway
                N Up Down         P.Up       P.Down
path:mmu05322  76  0   25 1.000000e+00 3.607857e-12
path:mmu05034 130 10   26 6.492511e-01 1.694044e-07
path:mmu03008  71  0   18 1.000000e+00 3.436244e-07
path:mmu05412  52 16    1 2.785106e-06 9.682854e-01
path:mmu04115  60  2   13 9.651619e-01 8.707340e-05
path:mmu04217 134  9   21 7.940145e-01 1.197793e-04
path:mmu05414  68 15    2 3.830266e-04 9.378625e-01
path:mmu05203 172  7   23 9.909773e-01 6.317987e-04
path:mmu05202 132  8   19 8.680539e-01 7.408914e-04
path:mmu05410  65 14    2 7.650457e-04 9.269001e-01
path:mmu00534  17  6    0 1.818698e-03 1.000000e+00
path:mmu03013 150  7   19 9.710062e-01 3.371750e-03
path:mmu05142  80  7   12 5.027586e-01 4.754486e-03
path:mmu04012  71 13    4 5.281088e-03 6.753477e-01
path:mmu04137  62  6   10 4.131237e-01 5.709701e-03
path:mmu05152 125 14   16 1.562344e-01 6.155321e-03
path:mmu03040 127  7   16 9.124164e-01 7.164024e-03
path:mmu04020 125 19    4 7.375150e-03 9.630814e-01
path:mmu05205 161 23   14 7.375432e-03 1.514618e-01
path:mmu04015 162 23    3 7.940031e-03 9.984658e-01</code></pre>
</div>
<div id="enrichment-of-gene-ontology-categories" class="section level2">
<h2>Enrichment of Gene Ontology categories</h2>
<p>Use <code>limma::goana</code> and <code>limma::topGO</code> to test for enrichment of differentially expressed genes in Gene Ontology categories.</p>
<pre class="r"><code>enrich_go_wt &lt;- goana(fit2, coef = &quot;DoxWT&quot;,
                      geneid = entrez, species = &quot;Mm&quot;)
topGO(enrich_go_wt, ontology = &quot;BP&quot;)</code></pre>
<pre><code>                                                       Term Ont     N   Up
GO:0090304                   nucleic acid metabolic process  BP  3705  764
GO:0016070                            RNA metabolic process  BP  3315  693
GO:0010467                                  gene expression  BP  3925  784
GO:0006396                                   RNA processing  BP   756  201
GO:0022613             ribonucleoprotein complex biogenesis  BP   387  119
GO:0042254                              ribosome biogenesis  BP   259   89
GO:0043170                  macromolecule metabolic process  BP  6446 1185
GO:0006139 nucleobase-containing compound metabolic process  BP  4135  804
GO:0034641     cellular nitrogen compound metabolic process  BP  4688  889
GO:0046483                    heterocycle metabolic process  BP  4225  810
GO:0006725     cellular aromatic compound metabolic process  BP  4258  813
GO:0007507                                heart development  BP   470   51
GO:0006364                                  rRNA processing  BP   177   63
GO:0008150                               biological_process  BP 13081 2074
GO:0034470                                 ncRNA processing  BP   314   94
GO:0060047                                heart contraction  BP   168   16
GO:0008016                  regulation of heart contraction  BP   136   13
GO:1901360        organic cyclic compound metabolic process  BP  4376  822
GO:0034097                             response to cytokine  BP   587  150
GO:0003015                                    heart process  BP   170   16
           Down         P.Up       P.Down
GO:0090304  615 5.725046e-19 9.997482e-01
GO:0016070  564 2.401108e-18 9.940679e-01
GO:0010467  682 1.096219e-15 9.828279e-01
GO:0006396  127 1.214908e-14 8.980481e-01
GO:0022613   51 1.369773e-13 9.980823e-01
GO:0042254   22 1.801469e-13 9.999983e-01
GO:0043170 1180 2.782282e-13 6.847707e-01
GO:0006139  724 4.147439e-13 9.730518e-01
GO:0034641  828 5.672903e-12 9.621104e-01
GO:0046483  752 9.985001e-12 9.147832e-01
GO:0006725  757 2.323034e-11 9.222017e-01
GO:0007507  144 9.994974e-01 6.030905e-11
GO:0006364   17 1.170328e-10 9.996583e-01
GO:0008150 2477 9.389315e-01 2.515467e-10
GO:0034470   44 2.636398e-10 9.858074e-01
GO:0060047   65 9.945028e-01 5.495593e-10
GO:0008016   56 9.890678e-01 5.582244e-10
GO:1901360  776 6.913125e-10 9.384157e-01
GO:0034097   97 7.647719e-10 9.035536e-01
GO:0003015   65 9.953871e-01 9.768464e-10</code></pre>
<pre class="r"><code>enrich_go_inter &lt;- goana(fit2, coef = &quot;Interaction&quot;,
                         geneid = entrez, species = &quot;Mm&quot;)
topGO(enrich_go_inter, ontology = &quot;BP&quot;)</code></pre>
<pre><code>                                           Term Ont   N Up Down
GO:0042254                  ribosome biogenesis  BP 259  8   53
GO:0022613 ribonucleoprotein complex biogenesis  BP 387 22   67
GO:0060047                    heart contraction  BP 168 41    6
GO:0003015                        heart process  BP 170 41    6
GO:0034470                     ncRNA processing  BP 314 21   50
GO:0048738    cardiac muscle tissue development  BP 208 45    5
GO:0003012                muscle system process  BP 294 56    6
GO:0008016      regulation of heart contraction  BP 136 34    6
GO:0060048           cardiac muscle contraction  BP 105 29    1
GO:0061337                   cardiac conduction  BP  45 18    0
GO:0086001 cardiac muscle cell action potential  BP  45 18    0
GO:0006334                  nucleosome assembly  BP  77  5   21
GO:1903522      regulation of blood circulation  BP 175 39   11
GO:0006364                      rRNA processing  BP 177  5   33
GO:0006941          striated muscle contraction  BP 132 32    2
GO:0086003      cardiac muscle cell contraction  BP  49 18    0
GO:0044057         regulation of system process  BP 361 62   18
GO:0042274   ribosomal small subunit biogenesis  BP  69  0   19
GO:0006396                       RNA processing  BP 756 65   87
GO:0016072               rRNA metabolic process  BP 204  5   35
                   P.Up       P.Down
GO:0042254 9.998643e-01 2.763336e-14
GO:0022613 9.835066e-01 5.764688e-14
GO:0060047 2.688492e-10 9.629910e-01
GO:0003015 3.965457e-10 9.658525e-01
GO:0034470 8.883100e-01 1.994496e-09
GO:0048738 2.390281e-09 9.978305e-01
GO:0003012 4.118983e-09 9.998955e-01
GO:0008016 4.617382e-09 8.768552e-01
GO:0060048 5.477954e-09 9.990864e-01
GO:0061337 7.146183e-09 1.000000e+00
GO:0086001 7.146183e-09 1.000000e+00
GO:0006334 7.846802e-01 9.280114e-09
GO:1903522 1.164167e-08 5.742726e-01
GO:0006364 9.993698e-01 2.461117e-08
GO:0006941 2.897487e-08 9.984797e-01
GO:0086003 3.478829e-08 1.000000e+00
GO:0044057 3.758163e-08 8.956282e-01
GO:0042274 1.000000e+00 3.967859e-08
GO:0006396 4.355818e-01 5.730293e-08
GO:0016072 9.999015e-01 8.278215e-08</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] limma_3.36.1        ggplot2_2.2.1       Biobase_2.40.0     
[4] BiocGenerics_0.26.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.17         compiler_3.5.0       pillar_1.2.3        
 [4] git2r_0.21.0         plyr_1.8.4           workflowr_1.0.1     
 [7] R.methodsS3_1.7.1    R.utils_2.6.0        tools_3.5.0         
[10] digest_0.6.15        bit_1.1-14           memoise_1.1.0       
[13] evaluate_0.10.1      RSQLite_2.1.1        tibble_1.4.2        
[16] gtable_0.2.0         pkgconfig_2.0.1      rlang_0.2.1         
[19] DBI_1.0.0            yaml_2.1.19          org.Mm.eg.db_3.6.0  
[22] stringr_1.3.1        knitr_1.20           IRanges_2.14.10     
[25] S4Vectors_0.18.3     stats4_3.5.0         rprojroot_1.3-2     
[28] bit64_0.9-7          grid_3.5.0           AnnotationDbi_1.42.1
[31] rmarkdown_1.10       blob_1.1.1           GO.db_3.6.0         
[34] magrittr_1.5         whisker_0.3-2        backports_1.1.2     
[37] scales_0.5.0         htmltools_0.3.6      colorspace_1.3-2    
[40] labeling_0.3         stringi_1.2.3        lazyeval_0.2.1      
[43] munsell_0.5.0        R.oo_1.22.0         </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.0.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
