<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="John Blischak" />


<title>Differential expression analysis for studies with a 2x2 factorial design</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DE analysis with limma</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/dc-bioc-limma">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential expression analysis for studies with a 2x2 factorial design</h1>
<h4 class="author"><em>John Blischak</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-05-23</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/jdblischak/dc-bioc-limma/tree/cc6353a03cb2d5c5c068d9f3e25f1cc3a2adbf0b" target="_blank">cc6353a</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    figure/.DS_Store
    Ignored:    figure/ch02/.DS_Store

Untracked files:
    Untracked:  _workflowr.yml

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/ch01.Rmd
    Modified:   analysis/ch02.Rmd
    Modified:   analysis/ch03.Rmd
    Modified:   analysis/ch04.Rmd
    Deleted:    analysis/chunks.R
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/ch04.html" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/a2fc9a7de10a2a1bd7c7aa3566c3bb7d47ba372c/analysis/ch04.Rmd" target="_blank">a2fc9a7</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
<td style="text-align:left;">
Minor edits to ch04 solutions
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/ch04.html" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/dc823825da0b6d5f60a9af5a31a2cb47fa735a7e/analysis/ch04.Rmd" target="_blank">dc82382</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
<td style="text-align:left;">
Minor updates to Ch 04 analysis.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/dc-bioc-limma/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/ch04.html" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/5db0175b96175b7b49c77fabff46836abc10a4a5/analysis/ch04.Rmd" target="_blank">5db0175</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
<td style="text-align:left;">
Organize Ch 04
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<p>Analayze microarray data from Zhang et al., 2012, which measured gene expression in hearts from wild type and Top2b null mice treated with doxorubicin or a control.</p>
<pre class="r"><code>library(&quot;Biobase&quot;)
library(&quot;ggplot2&quot;)
library(&quot;limma&quot;)
eset &lt;- readRDS(&quot;../data/ch04.rds&quot;)</code></pre>
<div id="studies-with-a-2x2-factorial-design-video" class="section level2">
<h2>Studies with a 2x2 factorial design (Video)</h2>
<p>Describe the scientific question, the experimental design, and the data collected for the 2x2 factorial study.</p>
</div>
<div id="expression-level-of-top2b" class="section level2">
<h2>Expression level of Top2b</h2>
<p>Create a boxplot of the expression level of Top2b to confirm the null mice have lower levels of Top2b expression. Use <code>pData</code> and <code>table</code> to access and explore the phenotype measurements to be used in the plot.</p>
<pre class="r"><code>head(fData(eset))</code></pre>
<pre><code>           probe   chr     start       end symbol            ensembl
1  A_55_P2051983 chr18  62753954  62753895 Spink7 ENSMUST00000076194
2   A_52_P169082 chr11  76031955  76032014  Dbil5                   
3 A_30_P01028193 chr14  47982421  47982480                          
4   A_52_P237997  chr3 156834624 156834683  Negr1                   
5   A_51_P414243  chr9 121890800 121890741 C85492 ENSMUST00000084743
6  A_55_P2136348  chr9 120003293 120003352   Ccr8 ENSMUST00000048777
        refseq entrez                                                name
1 NM_001001803 408198 serine peptidase inhibitor, Kazal type 7 (putative)
2    NM_021294  13168                   diazepam binding inhibitor-like 5
3                  NA                                                    
4    NM_177274 320840                         neuronal growth regulator 1
5    NM_153540 215494                           expressed sequence C85492
6    NM_007720  12776                    chemokine (C-C motif) receptor 8</code></pre>
<pre class="r"><code>top2b &lt;- exprs(eset)[fData(eset)$symbol == &quot;Top2b&quot;, ]
head(pData(eset))</code></pre>
<pre><code>                                                  title genotype treatment
WT.PBS.r1         WT Cardiomyocytes 16hr after PBS rep1       WT       PBS
WT.PBS.r2         WT Cardiomyocytes 16hr after PBS rep2       WT       PBS
WT.PBS.r3         WT Cardiomyocytes 16hr after PBS rep3       WT       PBS
WT.Dox.r1 WT Cardiomyocytes 16hr after doxorubicin rep1       WT       Dox
WT.Dox.r2 WT Cardiomyocytes 16hr after doxorubicin rep2       WT       Dox
WT.Dox.r3 WT Cardiomyocytes 16hr after doxorubicin rep3       WT       Dox
          rep
WT.PBS.r1  r1
WT.PBS.r2  r2
WT.PBS.r3  r3
WT.Dox.r1  r1
WT.Dox.r2  r2
WT.Dox.r3  r3</code></pre>
<pre class="r"><code>d &lt;- data.frame(pData(eset), top2b)
ggplot(d, aes(x = treatment, y = top2b, fill = genotype)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-1-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-1-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-1-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-1-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="visualize-gene-expression-distribution-with-boxplots" class="section level2">
<h2>Visualize gene expression distribution with boxplots</h2>
<p>Use boxplots as an alternative to density plots for checking the distribution of gene expression levels. Note pro (easier to see which sample belongs to each distribution) and con (hides any multimodality).</p>
<pre class="r"><code>boxplot(exprs(eset))</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-2-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-2-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-2-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-2-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>boxplot(log(exprs(eset)))</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-2-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-2-2.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>boxplot(log(exprs(eset)), las = 2)</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-2-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-2-3.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="perform-pca" class="section level2">
<h2>Perform PCA</h2>
<p>Use <code>prcomp</code> to calculate PCs and plot PC2 vs. PC1 to confirm that the samples are separated by genotype and treatment. Note that Top2b mice all cluster with untreated WT samples since they are resistant to the treatment.</p>
<pre class="r"><code>pca &lt;- prcomp(t(exprs(eset)), scale. = TRUE)
plot(pca)</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-3-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>d &lt;- data.frame(pData(eset), pca$x)
ggplot(d, aes(x = PC1, y = PC2, color = genotype, shape = treatment)) +
  geom_point()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-3-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-2.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-2.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-2.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>ggplot(d, aes(x = treatment, y = PC1, fill = genotype)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-3-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-3.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-3.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-3.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
</details>
<pre class="r"><code>ggplot(d, aes(x = treatment, y = PC2, fill = genotype)) +
  geom_boxplot()</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-3-4.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-3-4.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-3-4.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-3-4.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="differential-expression-for-2x2-factorial-designs-video" class="section level2">
<h2>Differential expression for 2x2 factorial designs (Video)</h2>
<p>Discuss how to construct a design and contrasts matrix for a 2x2 factorial design. Note how to create a combined factor and how to write the contrast for the interaction term.</p>
</div>
<div id="create-design-matrix-for-2x2-factorial-design" class="section level2">
<h2>Create design matrix for 2x2 factorial design</h2>
<p>Create a combined factor using <code>paste</code>. Use <code>model.matrix</code> to create a linear model with 4 binary variables (group-means parametrization).</p>
<pre class="r"><code>f &lt;- paste(pData(eset)$genotype, pData(eset)$treatment, sep = &quot;.&quot;)
design &lt;- model.matrix(~0 + f)
colnames(design) &lt;- sub(&quot;f&quot;, &quot;&quot;, colnames(design))
colSums(design)</code></pre>
<pre><code>Tob2b.Dox Tob2b.PBS    WT.Dox    WT.PBS 
        3         3         3         3 </code></pre>
</div>
<div id="create-contrasts-matrix-for-2x2-factorial-design" class="section level2">
<h2>Create contrasts matrix for 2x2 factorial design</h2>
<p>Use <code>limma::makeContrasts</code> to test the main effects and the interaction between genotype and treatment.</p>
<pre class="r"><code>cont_mat &lt;- makeContrasts(DoxWT = WT.Dox - WT.PBS,
                          DoxTop = Tob2b.Dox - Tob2b.PBS,
                          Interaction = (Tob2b.Dox - Tob2b.PBS) -
                                        (WT.Dox - WT.PBS),
                          levels = design)
cont_mat</code></pre>
<pre><code>           Contrasts
Levels      DoxWT DoxTop Interaction
  Tob2b.Dox     0      1           1
  Tob2b.PBS     0     -1          -1
  WT.Dox        1      0          -1
  WT.PBS       -1      0           1</code></pre>
</div>
<div id="fit-and-test-2x2-factorial-design" class="section level2">
<h2>Fit and test 2x2 factorial design</h2>
<p>Use <code>limma::lmFit</code>, <code>limma::contrasts.fit</code>, <code>limma::eBayes</code>, and <code>limma::decideTests</code> to fit and test the model. Use <code>limma::vennDiagram</code> to visualize overlap of differentially expressed genes.</p>
<pre class="r"><code>fit &lt;- lmFit(eset, design)
head(fit$coefficients)</code></pre>
<pre><code>    Tob2b.Dox   Tob2b.PBS      WT.Dox      WT.PBS
1   0.2975107   0.2945810   0.3539563   0.3620450
2   4.2250150   4.6055087   3.6762460   3.1691957
3   0.3001070   0.3196167   0.6157077   0.3739970
4   0.3012637   0.2955227   0.3532647   0.3616193
5 264.7589623 245.4630817 294.2404533 274.2917363
6   0.3032463   0.2958423   0.3525713   0.3611247</code></pre>
<pre class="r"><code>fit2 &lt;- contrasts.fit(fit, contrasts = cont_mat)
head(fit2$coefficients)</code></pre>
<pre><code>   Contrasts
           DoxWT       DoxTop Interaction
  1 -0.008088667  0.002929667  0.01101833
  2  0.507050333 -0.380493667 -0.88754400
  3  0.241710667 -0.019509667 -0.26122033
  4 -0.008354667  0.005741000  0.01409567
  5 19.948717000 19.295880667 -0.65283633
  6 -0.008553333  0.007404000  0.01595733</code></pre>
<pre class="r"><code>fit2 &lt;- eBayes(fit2)
results &lt;- decideTests(fit2)
summary(results)</code></pre>
<pre><code>   DoxWT DoxTop Interaction
-1  2311      0         320
0  55050  59304       58071
1   1944      1         914</code></pre>
<pre class="r"><code>vennDiagram(results)</code></pre>
<p><img src="figure/ch04.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-6-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/67f13a45f33e0667cc2e1c435a054cca66e86757/docs/figure/ch04.Rmd/unnamed-chunk-6-1.png" target="_blank">67f13a4</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-03-09
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/b2dc65335ad0fe8e6b398aeeaff7b85f64375925/docs/figure/ch04.Rmd/unnamed-chunk-6-1.png" target="_blank">b2dc653</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/dc-bioc-limma/blob/7f4d46b066c9a493f21149f02055b400adcd1fa1/docs/figure/ch04.Rmd/unnamed-chunk-6-1.png" target="_blank">7f4d46b</a>
</td>
<td style="text-align:left;">
John Blischak
</td>
<td style="text-align:left;">
2018-01-11
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="enrichment-analysis-video" class="section level2">
<h2>Enrichment analysis (Video)</h2>
<p>Describe common approaches for testing for enrichment of gene sets. Discuss Gene Onotlogy categories and KEGG pathways. Introduce Fisher’s Exact Test. Stress the importance of choosing the correct background set of genes and caution against over-interpretation (especially for directional effect). Mention at the end that there are more sophisticated algorithms such as <code>limma::camera</code>, <code>limma::roast</code>, and <code>goseq::goseq</code>.</p>
</div>
<div id="enrichment-of-kegg-pathways" class="section level2">
<h2>Enrichment of KEGG pathways</h2>
<p>Use <code>limma::kegga</code> and <code>limma::topKEGG</code> to test for enrichment of differentially expressed genes in KEGG pathways.</p>
<pre class="r"><code>entrez &lt;- fit2$genes$entrez
enrich_kegg_wt &lt;- kegga(fit2, coef = &quot;DoxWT&quot;,
                        geneid = entrez, species = &quot;Mm&quot;)
topKEGG(enrich_kegg_wt)</code></pre>
<pre><code>                                                             Pathway   N
path:mmu05322                           Systemic lupus erythematosus 110
path:mmu05034                                             Alcoholism 165
path:mmu03008                      Ribosome biogenesis in eukaryotes  77
path:mmu05203                                   Viral carcinogenesis 212
path:mmu05412 Arrhythmogenic right ventricular cardiomyopathy (ARVC)  72
path:mmu04137                                     Mitophagy - animal  63
path:mmu04261                 Adrenergic signaling in cardiomyocytes 145
path:mmu04022                             cGMP-PKG signaling pathway 166
path:mmu05414                           Dilated cardiomyopathy (DCM)  88
path:mmu04217                                            Necroptosis 158
path:mmu04971                                 Gastric acid secretion  74
path:mmu04144                                            Endocytosis 262
path:mmu05410                      Hypertrophic cardiomyopathy (HCM)  83
path:mmu05202                Transcriptional misregulation in cancer 174
path:mmu04012                                 ErbB signaling pathway  84
path:mmu03040                                            Spliceosome 129
path:mmu03013                                          RNA transport 166
path:mmu04066                                HIF-1 signaling pathway 103
path:mmu04115                                  p53 signaling pathway  67
path:mmu04919                      Thyroid hormone signaling pathway 115
              Up Down         P.Up       P.Down
path:mmu05322 25    1 1.275512e-10 9.997136e-01
path:mmu05034 29   12 2.754611e-09 5.089928e-01
path:mmu03008 19    2 5.184560e-09 9.767149e-01
path:mmu05203 33   11 5.356776e-09 8.957936e-01
path:mmu05412  0   20 1.000000e+00 7.142217e-08
path:mmu04137 15    7 3.628333e-07 1.572432e-01
path:mmu04261  5   27 8.642262e-01 2.843226e-06
path:mmu04022  4   29 9.717775e-01 4.646843e-06
path:mmu05414  0   19 1.000000e+00 9.524657e-06
path:mmu04217 21   14 4.219986e-05 2.306917e-01
path:mmu04971  2   16 8.948275e-01 4.706902e-05
path:mmu04144 29   20 5.284433e-05 4.005582e-01
path:mmu05410  0   17 1.000000e+00 5.716438e-05
path:mmu05202 22   10 6.030451e-05 7.996927e-01
path:mmu04012  5   17 4.206570e-01 6.704920e-05
path:mmu03040 18   11 7.835780e-05 3.079244e-01
path:mmu03013 21    8 8.782808e-05 9.111483e-01
path:mmu04066  8   19 1.491963e-01 9.619306e-05
path:mmu04115 12    2 1.127063e-04 9.565283e-01
path:mmu04919  7   20 3.627170e-01 1.484823e-04</code></pre>
<pre class="r"><code>enrich_kegg_inter &lt;- kegga(fit2, coef = &quot;Interaction&quot;,
                        geneid = entrez, species = &quot;Mm&quot;)
topKEGG(enrich_kegg_inter)</code></pre>
<pre><code>                                                             Pathway   N
path:mmu03008                      Ribosome biogenesis in eukaryotes  77
path:mmu05322                           Systemic lupus erythematosus 110
path:mmu05034                                             Alcoholism 165
path:mmu05203                                   Viral carcinogenesis 212
path:mmu04919                      Thyroid hormone signaling pathway 115
path:mmu00230                                      Purine metabolism 175
path:mmu05414                           Dilated cardiomyopathy (DCM)  88
path:mmu05412 Arrhythmogenic right ventricular cardiomyopathy (ARVC)  72
path:mmu05212                                      Pancreatic cancer  75
path:mmu05220                               Chronic myeloid leukemia  76
path:mmu05205                                Proteoglycans in cancer 199
path:mmu04930                              Type II diabetes mellitus  48
path:mmu04211                           Longevity regulating pathway  90
path:mmu04261                 Adrenergic signaling in cardiomyocytes 145
path:mmu05222                                 Small cell lung cancer  92
path:mmu04210                                              Apoptosis 134
path:mmu05213                                     Endometrial cancer  58
path:mmu04933   AGE-RAGE signaling pathway in diabetic complications 100
path:mmu05217                                   Basal cell carcinoma  63
path:mmu01522                                   Endocrine resistance  93
              Up Down         P.Up       P.Down
path:mmu03008  0   10 1.0000000000 1.007965e-08
path:mmu05322  0   10 1.0000000000 3.200229e-07
path:mmu05034  5   11 0.5239903502 1.823095e-06
path:mmu05203  3   12 0.9487940322 3.363101e-06
path:mmu04919 11    2 0.0004774097 3.663623e-01
path:mmu00230 14    3 0.0005362122 3.077808e-01
path:mmu05414  9    0 0.0009610474 1.000000e+00
path:mmu05412  8    0 0.0010648869 1.000000e+00
path:mmu05212  3    5 0.3710324131 1.411325e-03
path:mmu05220  4    5 0.1779928540 1.497676e-03
path:mmu05205 14    5 0.0018741551 7.000390e-02
path:mmu04930  6    1 0.0024581255 4.159308e-01
path:mmu04211  3    5 0.4867180093 3.156205e-03
path:mmu04261 11    1 0.0031788883 8.049817e-01
path:mmu05222  1    5 0.9342324019 3.471129e-03
path:mmu04210  6    6 0.1928783824 3.643308e-03
path:mmu05213  3    4 0.2361147774 3.816237e-03
path:mmu04933  6    5 0.0698159306 4.959486e-03
path:mmu05217  4    4 0.1092379866 5.137883e-03
path:mmu01522  8    3 0.0053948301 8.431867e-02</code></pre>
</div>
<div id="enrichment-of-gene-ontology-categories" class="section level2">
<h2>Enrichment of Gene Ontology categories</h2>
<p>Use <code>limma::goana</code> and <code>limma::topGO</code> to test for enrichment of differentially expressed genes in Gene Ontology categories.</p>
<pre class="r"><code>enrich_go_wt &lt;- goana(fit2, coef = &quot;DoxWT&quot;,
                      geneid = entrez, species = &quot;Mm&quot;)
topGO(enrich_go_wt, ontology = &quot;BP&quot;)</code></pre>
<pre><code>                                                       Term Ont    N  Up
GO:0090304                   nucleic acid metabolic process  BP 4256 411
GO:0016070                            RNA metabolic process  BP 3810 373
GO:0044260         cellular macromolecule metabolic process  BP 7072 577
GO:0006139 nucleobase-containing compound metabolic process  BP 4786 426
GO:0034641     cellular nitrogen compound metabolic process  BP 5404 460
GO:0046483                    heterocycle metabolic process  BP 4882 427
GO:0043170                  macromolecule metabolic process  BP 7739 598
GO:0006725     cellular aromatic compound metabolic process  BP 4945 429
GO:0010467                                  gene expression  BP 4539 403
GO:0006807              nitrogen compound metabolic process  BP 5725 474
GO:1901360        organic cyclic compound metabolic process  BP 5109 432
GO:0006396                                   RNA processing  BP  742 113
GO:0044237                       cellular metabolic process  BP 8723 633
GO:0022613             ribonucleoprotein complex biogenesis  BP  366  71
GO:0044238                        primary metabolic process  BP 8743 626
GO:0034645      cellular macromolecule biosynthetic process  BP 4075 344
GO:0071704              organic substance metabolic process  BP 9212 645
GO:0051252              regulation of RNA metabolic process  BP 3121 280
GO:0032774                         RNA biosynthetic process  BP 3092 278
GO:0097659             nucleic acid-templated transcription  BP 3078 277
           Down         P.Up       P.Down
GO:0090304  331 1.504047e-37 8.626750e-03
GO:0016070  306 1.468665e-34 1.970377e-03
GO:0044260  589 2.147486e-34 9.800194e-09
GO:0006139  393 1.214879e-30 5.157099e-05
GO:0034641  453 4.594157e-29 8.906329e-07
GO:0046483  406 5.642472e-29 1.051022e-05
GO:0043170  630 7.077190e-29 8.377610e-08
GO:0006725  409 2.423666e-28 1.647989e-05
GO:0010467  378 2.692286e-28 2.196723e-05
GO:0006807  472 2.722933e-27 3.295379e-06
GO:1901360  420 4.358967e-26 2.178695e-05
GO:0006396   65 7.616058e-24 3.022888e-02
GO:0044237  731 3.723539e-23 1.744190e-12
GO:0022613   26 3.096042e-21 4.775639e-01
GO:0044238  725 4.779276e-21 2.905773e-11
GO:0034645  334 1.552262e-19 2.752467e-04
GO:0071704  765 2.262336e-19 1.992779e-12
GO:0051252  254 4.844028e-19 2.593424e-03
GO:0032774  244 5.051745e-19 1.343578e-02
GO:0097659  242 5.273962e-19 1.615435e-02</code></pre>
<pre class="r"><code>enrich_go_inter &lt;- goana(fit2, coef = &quot;Interaction&quot;,
                         geneid = entrez, species = &quot;Mm&quot;)
topGO(enrich_go_inter, ontology = &quot;BP&quot;)</code></pre>
<pre><code>                                                    Term Ont    N  Up Down
GO:0034470                              ncRNA processing  BP  300  10   20
GO:0042254                           ribosome biogenesis  BP  249   4   18
GO:0006334                           nucleosome assembly  BP   90   1   12
GO:0006396                                RNA processing  BP  742  28   30
GO:0034728                       nucleosome organization  BP  114   2   13
GO:0034660                       ncRNA metabolic process  BP  402  12   22
GO:0071103                       DNA conformation change  BP  174   3   15
GO:0031497                            chromatin assembly  BP  106   1   12
GO:0022613          ribonucleoprotein complex biogenesis  BP  366  11   20
GO:0016043               cellular component organization  BP 5401 230   67
GO:0006364                               rRNA processing  BP  169   3   14
GO:0060047                             heart contraction  BP  187  24    2
GO:0003015                                 heart process  BP  190  24    2
GO:0016072                        rRNA metabolic process  BP  174   3   14
GO:0090304                nucleic acid metabolic process  BP 4256 143   81
GO:0061337                            cardiac conduction  BP   45  12    0
GO:0086003               cardiac muscle cell contraction  BP   46  12    0
GO:0006333             chromatin assembly or disassembly  BP  127   2   12
GO:0071840 cellular component organization or biogenesis  BP 5552 232   78
GO:0030490                        maturation of SSU-rRNA  BP   47   0    8
                   P.Up       P.Down
GO:0034470 4.276026e-01 4.956310e-11
GO:0042254 9.468523e-01 1.280676e-10
GO:0006334 9.380859e-01 1.522812e-10
GO:0006396 1.404747e-01 1.829287e-10
GO:0034728 8.649106e-01 1.912084e-10
GO:0034660 5.655911e-01 2.325658e-10
GO:0071103 9.018931e-01 4.099080e-10
GO:0031497 9.622922e-01 1.058346e-09
GO:0022613 5.571261e-01 1.637937e-09
GO:0016043 2.509447e-09 5.204254e-02
GO:0006364 8.908148e-01 2.688905e-09
GO:0060047 2.742485e-09 5.797084e-01
GO:0003015 3.784836e-09 5.883840e-01
GO:0016072 9.018931e-01 3.928343e-09
GO:0090304 9.322263e-02 4.457810e-09
GO:0061337 6.378454e-09 1.000000e+00
GO:0086003 8.393293e-09 1.000000e+00
GO:0006333 9.016483e-01 8.547694e-09
GO:0071840 1.190080e-08 1.306441e-03
GO:0030490 1.000000e+00 2.600262e-08</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X Yosemite 10.10.5

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] limma_3.30.13       ggplot2_2.2.1       Biobase_2.34.0     
[4] BiocGenerics_0.20.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14         pillar_1.2.2         git2r_0.21.0        
 [4] plyr_1.8.4           workflowr_1.0.1      R.methodsS3_1.7.1   
 [7] R.utils_2.6.0        tools_3.3.3          digest_0.6.13       
[10] bit_1.1-12           memoise_1.1.0        evaluate_0.10.1     
[13] RSQLite_2.0          tibble_1.4.2         gtable_0.2.0        
[16] pkgconfig_2.0.1      rlang_0.2.0          DBI_0.8             
[19] yaml_2.1.16          org.Mm.eg.db_3.4.0   stringr_1.3.0       
[22] knitr_1.20           IRanges_2.8.2        S4Vectors_0.12.2    
[25] stats4_3.3.3         rprojroot_1.3-2      bit64_0.9-7         
[28] grid_3.3.3           AnnotationDbi_1.36.2 rmarkdown_1.9.12    
[31] GO.db_3.4.0          blob_1.1.1           magrittr_1.5        
[34] whisker_0.3-2        backports_1.1.2      scales_0.5.0        
[37] htmltools_0.3.6      colorspace_1.3-2     labeling_0.3        
[40] stringi_1.1.7        lazyeval_0.2.1       munsell_0.4.3       
[43] R.oo_1.22.0         </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.0.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
