---
title: "Differential Expression with limma"
output: html_document
---

```{r include=FALSE}
library("knitr")
opts_chunk$set(eval = TRUE)
```

## Chapter 1: Differential Expression Testing

### Lesson 1: Genomics experiments

```{r cll}
library(CLL)
data("sCLLex")
class(sCLLex)
dim(sCLLex)
head(pData(sCLLex))
head(fData(sCLLex))
boxplot(exprs(sCLLex)[1, ] ~ pData(sCLLex)$Disease)
x <- exprs(sCLLex)
f <- fData(sCLLex)
p <- pData(sCLLex)
library(hgu95av2.db)
columns(hgu95av2.db)
# select(x, keys, columns, keytype, ...) Couldn't get multiVals argument to
# work. Seems to be ignored. I tried "first", "CharacterList", and a custom
# function to select the last possible value. None made any difference.
hope <- select(hgu95av2.db, keys = rownames(x),
               columns = c("SYMBOL", "ENTREZID", "MAP", "GENENAME"))
hope <- hope[!duplicated(hope$PROBEID), ]
stopifnot(hope$PROBEID == rownames(x))
colnames(hope) <- c("probe", "symbol", "entrez", "chrom", "name")
rownames(hope) <- hope$probe
hope[["probe"]] <- NULL
f <- hope
boxplot(x[1, ] ~ p[, "Disease"], main = f[1, "symbol"])
library(Biobase)
eset <- ExpressionSet(assayData = x,
                      phenoData = AnnotatedDataFrame(p),
                      featureData = AnnotatedDataFrame(f))
save(list = c("x", "f", "p"), file = "../data/cll.RData")
saveRDS(eset, file = "../data/cll-eset.rds")
dim(eset)

# Subset to only include the 1000th gene (row) and the first 10 samples
eset_sub <- eset[1000, 1:10]

# Check the dimensions of the subset 
dim(eset_sub)

# Create a boxplot of the first gene in eset_sub
boxplot(exprs(eset_sub)[1, ] ~ pData(eset_sub)[, "Disease"], main = fData(eset_sub)[1, "symbol"])

# Create design matrix for CLL study
design <- model.matrix(~Disease, data = pData(eset))

# Count the number of samples modeled by each coefficient
colSums(design)

# Load package
library(limma)

# Fit the model
fit <- lmFit(eset, design)

# Calculate the t-statistics
fit <- eBayes(fit)

# Summarize results
results <- decideTests(fit[, "Diseasestable"])
summary(results)


# group-means

# Create design matrix with no intercept
design <- model.matrix(~0 + Disease, data = pData(eset))

# Count the number of samples modeled by each coefficient
colSums(design)

# Load package
library(limma)

# Create a contrasts matrix
cm <- makeContrasts(status = Diseaseprogres. - Diseasestable,
                    levels = design)

# View the contrasts matrix
cm

# Load package
library(limma)

# Fit the model
fit <- lmFit(eset, design)

# Fit the contrasts
fit2 <- contrasts.fit(fit, contrasts = cm)

# Calculate the t-statistics for the contrasts
fit2 <- eBayes(fit2)

# Summarize results
results <- decideTests(fit2)
summary(results)
```

```{r vdx}
library(breastCancerVDX)
data("vdx")
class(vdx)
dim(vdx)
head(pData(vdx))
head(fData(vdx))
x <- exprs(vdx)
f <- fData(vdx)
p <- pData(vdx)
# Recode er as 0 = negative and 1 = positive
p[, "er"] <- ifelse(p[, "er"] == 0, "negative", "positive")
png("../figure/ch01/boxplot.png")
boxplot(x[1, ] ~ p[, "er"], main = f[1, "Gene.symbol"])
dev.off()
eset <- ExpressionSet(assayData = x,
                      phenoData = AnnotatedDataFrame(p),
                      featureData = AnnotatedDataFrame(f))
dim(eset)
boxplot(exprs(eset)[1, ] ~ pData(eset)[, "er"],
        main = fData(eset)[1, "Gene.symbol"])
design <- model.matrix(~er, data = pData(eset))
head(design, 3)
colSums(design)
table(pData(eset)[, "er"])

library(limma)

fit <- lmFit(eset, design)
head(fit$coefficients, 3)
fit <- eBayes(fit)
head(fit$t, 3)
results <- decideTests(fit[, "erpositive"])
summary(results)

# group-means
design <- model.matrix(~0 + er, data = pData(eset))
head(design)
colSums(design)

library(limma)
cm <- makeContrasts(status = erpositive - ernegative,
                    levels = design)
cm

fit <- lmFit(eset, design)
head(fit$coefficients)
fit2 <- contrasts.fit(fit, contrasts = cm)
head(fit2$coefficients)
fit2 <- eBayes(fit2)
results <- decideTests(fit2)
summary(results)
```

#### VideoExercise: Genomics Experiments

#### MultipleChoiceExercise: Genomic Technologies

#### Normal Exercise: 

### Lesson 2: Managing gene expression data

### Lesson 3: Linear modelling with limma

#### VideoExercise: Linear modelling with limma

#### NormalExercise: Specify the model

#### NormalExercise: Perform the hypothesis test

## Chapter 2: Flexible models for genomics studies

#### Lesson 1: Using group-means parametrization

#### VideoExercise: Using group-means parametrization

#### NormalExercise: Specify the model

#### NormalExercise: Specify the contrasts

#### NormalExercise: Test the model

#### Lesson 2: Using group-means parametrization for studies with more than 2 groups

#### VideoExercise: Using group-means parametrization

```{r stemhypoxia}
library(stemHypoxia)
data(stemHypoxia)
dim(M)
class(M)
M[1:5, 1:5]
f <- M[, 1:2]
rownames(f) <- f$Gene_ID
rownames(M) <- f$Gene_ID
M[, 1:2] <- NULL
dim(design)
class(design)
head(design)
rownames(design) <- colnames(M)

eset <- ExpressionSet(assayData = as.matrix(M),
                      phenoData = AnnotatedDataFrame(design),
                      featureData = AnnotatedDataFrame(f))
table(design$oxygen, design$time)
# Subset to only incldue the 5 days timepoint
eset <- eset[, pData(eset)$time == 5]
dim(eset)
# Clean up names
library(stringr)
pData(eset)[, "oxygen"] <- sprintf("ox%02d", pData(eset)[, "oxygen"])
pData(eset)[, "rep"] <- sprintf("r%d",
                                as.integer(str_sub(pData(eset)[, "samplename"], -1, -1)))
pData(eset)[, c("time", "samplename")] <- NULL
colnames(eset) <- paste(pData(eset)[, "oxygen"], pData(eset)[, "rep"], sep = "_")
head(pData(eset))

dim(eset)
table(pData(eset)[, "oxygen"])
```

```{r leukemiasEset}
library(leukemiasEset)
data("leukemiasEset")
?leukemiasEset
eset <- leukemiasEset
dim(eset)
head(fData(eset))
featureData(eset) <- AnnotatedDataFrame(data.frame(ensembl = rownames(exprs(eset)),
                                                   stringsAsFactors = FALSE))
head(fData(eset))
exprs(eset)[1:5, 1:5]
head(pData(eset))
table(pData(eset)[, "LeukemiaType"])
# Subset to only include ALL, AML, and CML
eset <- eset[, pData(eset)[, "LeukemiaType"] %in% c("ALL","AML", "CML")]
dim(eset)
# Clean up names
phenoData(eset) <- AnnotatedDataFrame(data.frame(type = as.character(pData(eset)[, "LeukemiaType"]),
                                                   stringsAsFactors = FALSE))
head(pData(eset))
exprs(eset)[1:5, 1:5]
colnames(eset) <- sprintf("sample_%02d", 1:ncol(eset))
exprs(eset)[1:5, 1:5]
dim(eset)
head(pData(eset), 3)
table(pData(eset)[, "type"])

design <- model.matrix(~0 + type, data = pData(eset))
head(design, 3)
colSums(design)

library(limma)
# * Tests:
#     * AML v. ALL: $\beta_2 - \beta_1 = 0$
#     * CML v. ALL: $\beta_3 - \beta_1 = 0$
#     * CML v. AML: $\beta_3 - \beta_2 = 0$
cm <- makeContrasts(AMLvALL = typeAML - typeALL,
                    CMLvALL = typeCML - typeALL,
                    CMLvAML = typeCML - typeAML,
                    levels = design)
cm

library(limma)
# Fit coefficients
fit <- lmFit(eset, design)
# Fit contrasts
fit2 <- contrasts.fit(fit, contrasts = cm)
# Calculate t-statistics
fit2 <- eBayes(fit2)
# Summarize results
results <- decideTests(fit2)
summary(results)
```

#### NormalExercise: Specify the model

#### NormalExercise: Specify the contrasts

#### NormalExercise: Test the model

#### Lesson 3: Using group-means parametrization

#### VideoExercise: Using group-means parametrization

#### NormalExercise: Specify the model

#### NormalExercise: Specify the contrasts

#### NormalExercise: Test the model

## Chapter 3: Pre- and post-processing

```{r}
# BiocInstaller::biocLite("bladderbatch")
library(bladderbatch)
data(bladderdata)
bladderEset
eset <- bladderEset
head(pData(eset))
head(fData(eset))
exprs(eset)[1:5, 1:5]
table(pData(eset)[, c("outcome", "batch")])
library("limma")
eset <- eset[, pData(eset)$outcome %in% c("sTCC-CIS", "sTCC+CIS")]
plotMDS(eset, labels = pData(eset)$batch, gene.selection = "common")
plotMDS(eset, labels = pData(eset)$outcome, gene.selection = "common")
exprs(eset) <- removeBatchEffect(eset, batch = pData(eset)$batch)
plotMDS(eset, labels = pData(eset)$batch, gene.selection = "common")
plotMDS(eset, labels = pData(eset)$outcome, gene.selection = "common")
```

Recommend reading [Leek et al., 2010](Tackling the widespread and critical impact of batch effects in high-throughput data)

```{r}
# BiocInstaller::biocLite("HarmanData")
library(HarmanData)
data(IMR90)
imr90.info
table(imr90.info)
plotMDS(imr90.data, labels = imr90.info$Treatment, gene.selection = "common")
plotMDS(imr90.data, labels = imr90.info$Batch, gene.selection = "common")
imr90_clean <- removeBatchEffect(imr90.data, batch = imr90.info$Batch)
plotMDS(imr90_clean, labels = imr90.info$Treatment)
plotMDS(imr90_clean, labels = imr90.info$Batch)

data(NPM)
npm.info
table(npm.info)
plotMDS(npm.data, labels = npm.info$Treatment, gene.selection = "common")
plotMDS(npm.data, labels = npm.info$Batch, gene.selection = "common")
npm_clean <- removeBatchEffect(npm.data, batch = npm.info$Batch, design = model.matrix(~Treatment, data = npm.info))
plotMDS(npm_clean, labels = npm.info$Treatment)
plotMDS(npm_clean, labels = npm.info$Batch)

data(OLF)
olf.info
table(olf.info)
plotMDS(olf.data, labels = olf.info$Treatment, gene.selection = "common")
plotMDS(olf.data, labels = olf.info$Batch, gene.selection = "common")
olf_clean <- removeBatchEffect(olf.data, batch = olf.info$Batch, design = model.matrix(~Treatment, data = olf.info))
olf_clean <- removeBatchEffect(olf.data, batch = olf.info$Batch)
plotMDS(olf_clean, labels = olf.info$Treatment, gene.selection = "common", col = olf.info$Treatment)
plotMDS(olf_clean, labels = olf.info$Batch, gene.selection = "common", col = olf.info$Batch)
```

OLF study: https://particleandfibretoxicology.biomedcentral.com/articles/10.1186/1743-8977-10-54
OLF GEO: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45322

## Chapter 4: Case study: 

## Other data sets

* [ALL](https://www.bioconductor.org/packages/release/data/experiment/html/ALL.html)
* [yeastCC](https://bioconductor.org/packages/release/data/experiment/html/yeastCC.html)
* [gaschYHS](https://www.bioconductor.org/packages/release/data/experiment/html/gaschYHS.html)
* Multiple yeast mutants and multiple stress treatments - [Gasch et al., 2001](https://www.ncbi.nlm.nih.gov/pubmed/11598186) and [companion website](http://www-genome.stanford.edu/mec1/)
* 2x2 factorial myostatin mouse - [Dilger et al., 2010](https://www.ncbi.nlm.nih.gov/pubmed/22443951) and [GEO entry](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE27577) - 2 color arrays
* 2x2 factorial protein/infection effect on fetal brain - [Starr et al., 2016 GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE57115)
* [bladderbatch](https://www.bioconductor.org/packages/release/data/experiment/html/bladderbatch.html)
* [HarmanData](https://www.bioconductor.org/packages/release/data/experiment/html/HarmanData.html)
* [Arabidopsis thaliana 2x2](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53990)
* [Genotype and time of day shape the Populus drought response](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15242)
